diff -up linux-5.15/arch/arm64/boot/dts/rockchip/rk3399-pinebook-pro.dts.60~ linux-5.15/arch/arm64/boot/dts/rockchip/rk3399-pinebook-pro.dts
--- linux-5.15/arch/arm64/boot/dts/rockchip/rk3399-pinebook-pro.dts.60~	2021-11-25 20:12:02.649659672 +0100
+++ linux-5.15/arch/arm64/boot/dts/rockchip/rk3399-pinebook-pro.dts	2021-11-25 20:13:10.705935209 +0100
@@ -385,6 +385,11 @@
 	};
 };
 
+&cdn_dp {
+	status = "okay";
+	extcon = <&fusb0>;
+};
+
 &cpu_b0 {
 	cpu-supply = <&vdd_cpu_b>;
 };
@@ -717,6 +722,9 @@
 				<PDO_FIXED(5000, 1400, PDO_FIXED_USB_COMM)>;
 			try-power-role = "sink";
 
+			extcon-cables = <1 2 5 6 9 10 12 44>;
+			typec-altmodes = <0xff01 1 0x001c0000 1>;
+
 			ports {
 				#address-cells = <1>;
 				#size-cells = <0>;
@@ -983,6 +991,7 @@
 };
 
 &tcphy0 {
+	extcon = <&fusb0>;
 	status = "okay";
 };
 
