--- linux-4.6/arch/arm64/Kconfig.0035~	2016-06-02 15:36:43.494016927 +0200
+++ linux-4.6/arch/arm64/Kconfig	2016-06-02 15:36:45.403003582 +0200
@@ -2,6 +2,7 @@ config ARM64
 	def_bool y
 	select ACPI_CCA_REQUIRED if ACPI
 	select ACPI_GENERIC_GSI if ACPI
+	select ACPI_PCI_HOST_GENERIC if ACPI
 	select ACPI_REDUCED_HARDWARE_ONLY if ACPI
 	select ARCH_HAS_DEVMEM_IS_ALLOWED
 	select ARCH_HAS_ATOMIC64_DEC_IF_POSITIVE
--- linux-4.6/arch/arm64/kernel/pci.c.0035~	2016-06-02 15:36:45.403003582 +0200
+++ linux-4.6/arch/arm64/kernel/pci.c	2016-06-02 15:37:37.719638763 +0200
@@ -88,12 +88,3 @@ int pcibus_to_node(struct pci_bus *bus)
 EXPORT_SYMBOL(pcibus_to_node);
 
 #endif
-
-#ifdef CONFIG_ACPI
-/* Root bridge scanning */
-struct pci_bus *pci_acpi_scan_root(struct acpi_pci_root *root)
-{
-	/* TODO: Should be revisited when implementing PCI on ACPI */
-	return NULL;
-}
-#endif
