--- linux-4.5/arch/arm64/Kconfig.0035~	2016-04-03 10:30:59.934786425 +0200
+++ linux-4.5/arch/arm64/Kconfig	2016-04-03 10:31:01.638070081 +0200
@@ -2,6 +2,7 @@ config ARM64
 	def_bool y
 	select ACPI_CCA_REQUIRED if ACPI
 	select ACPI_GENERIC_GSI if ACPI
+	select ACPI_PCI_HOST_GENERIC if ACPI
 	select ACPI_REDUCED_HARDWARE_ONLY if ACPI
 	select ARCH_HAS_DEVMEM_IS_ALLOWED
 	select ARCH_HAS_ATOMIC64_DEC_IF_POSITIVE
--- linux-4.5/arch/arm64/kernel/pci.c.0035~	2016-04-03 10:31:01.638070081 +0200
+++ linux-4.5/arch/arm64/kernel/pci.c	2016-04-03 10:31:34.936216826 +0200
@@ -78,12 +78,3 @@ int raw_pci_write(unsigned int domain, u
 {
 	return -ENXIO;
 }
-
-#ifdef CONFIG_ACPI
-/* Root bridge scanning */
-struct pci_bus *pci_acpi_scan_root(struct acpi_pci_root *root)
-{
-	/* TODO: Should be revisited when implementing PCI on ACPI */
-	return NULL;
-}
-#endif
