diff -up linux-5.0/drivers/misc/vmw_balloon.c.44~ linux-5.0/drivers/misc/vmw_balloon.c
--- linux-5.0/drivers/misc/vmw_balloon.c.44~	2019-03-04 15:27:09.759740794 +0100
+++ linux-5.0/drivers/misc/vmw_balloon.c	2019-03-04 15:28:04.324597379 +0100
@@ -1363,7 +1363,7 @@ static void vmballoon_work(struct work_s
 	 * stopped while the system transitions to/from sleep/hibernation.
 	 */
 	queue_delayed_work(system_freezable_wq,
-			   dwork, round_jiffies_relative(HZ));
+			   dwork, round_jiffies_relative(10*HZ));
 
 }
 
diff -up linux-5.0/drivers/pci/pci.c.44~ linux-5.0/drivers/pci/pci.c
--- linux-5.0/drivers/pci/pci.c.44~	2019-03-04 00:21:29.000000000 +0100
+++ linux-5.0/drivers/pci/pci.c	2019-03-04 15:27:09.759740794 +0100
@@ -61,7 +61,7 @@ struct pci_pme_device {
 	struct pci_dev *dev;
 };
 
-#define PME_TIMEOUT 1000 /* How long between PME checks */
+#define PME_TIMEOUT 4000 /* How long between PME checks */
 
 static void pci_dev_d3_sleep(struct pci_dev *dev)
 {
