diff -up linux-4.20/drivers/misc/vmw_balloon.c.33~ linux-4.20/drivers/misc/vmw_balloon.c
--- linux-4.20/drivers/misc/vmw_balloon.c.33~	2018-12-24 14:35:58.168980706 +0100
+++ linux-4.20/drivers/misc/vmw_balloon.c	2018-12-24 14:39:46.097033560 +0100
@@ -1363,7 +1363,7 @@ static void vmballoon_work(struct work_s
 	 * stopped while the system transitions to/from sleep/hibernation.
 	 */
 	queue_delayed_work(system_freezable_wq,
-			   dwork, round_jiffies_relative(HZ));
+			   dwork, round_jiffies_relative(10*HZ));
 
 }
 
diff -up linux-4.20/drivers/pci/pci.c.33~ linux-4.20/drivers/pci/pci.c
--- linux-4.20/drivers/pci/pci.c.33~	2018-12-24 00:55:59.000000000 +0100
+++ linux-4.20/drivers/pci/pci.c	2018-12-24 14:35:58.169980720 +0100
@@ -61,7 +61,7 @@ struct pci_pme_device {
 	struct pci_dev *dev;
 };
 
-#define PME_TIMEOUT 1000 /* How long between PME checks */
+#define PME_TIMEOUT 4000 /* How long between PME checks */
 
 static void pci_dev_d3_sleep(struct pci_dev *dev)
 {
