diff -up linux-4.20/arch/x86/kernel/tsc.c.37~ linux-4.20/arch/x86/kernel/tsc.c
--- linux-4.20/arch/x86/kernel/tsc.c.37~	2018-12-24 14:42:03.205868949 +0100
+++ linux-4.20/arch/x86/kernel/tsc.c	2018-12-24 14:43:25.875160716 +0100
@@ -1503,6 +1503,9 @@ unsigned long calibrate_delay_is_known(v
 	if (!constant_tsc || !mask)
 		return 0;
 
+	if (cpu != 0)
+		return cpu_data(0).loops_per_jiffy;
+
 	sibling = cpumask_any_but(mask, cpu);
 	if (sibling < nr_cpu_ids)
 		return cpu_data(sibling).loops_per_jiffy;
