diff -up linux-5.1/arch/x86/kernel/cpu/intel.c.49~ linux-5.1/arch/x86/kernel/cpu/intel.c
--- linux-5.1/arch/x86/kernel/cpu/intel.c.49~	2019-05-07 17:36:54.729087361 +0200
+++ linux-5.1/arch/x86/kernel/cpu/intel.c	2019-05-07 17:39:10.005775786 +0200
@@ -608,12 +608,12 @@ static void init_intel_energy_perf(struc
 		return;
 
 	rdmsrl(MSR_IA32_ENERGY_PERF_BIAS, epb);
-	if ((epb & 0xF) != ENERGY_PERF_BIAS_PERFORMANCE)
+	if ((epb & 0xF) >= ENERGY_PERF_BIAS_NORMAL)
 		return;
 
-	pr_info_once("ENERGY_PERF_BIAS: Set to 'normal', was 'performance'\n");
+	pr_info_once("ENERGY_PERF_BIAS: Set to 'performance', was 'normal'\n");
 	pr_info_once("ENERGY_PERF_BIAS: View and update with x86_energy_perf_policy(8)\n");
-	epb = (epb & ~0xF) | ENERGY_PERF_BIAS_NORMAL;
+	epb = (epb & ~0xF) | ENERGY_PERF_BIAS_PERFORMANCE;
 	wrmsrl(MSR_IA32_ENERGY_PERF_BIAS, epb);
 }
 
