diff -up linux-5.10/drivers/tty/sysrq.c.66~ linux-5.10/drivers/tty/sysrq.c
--- linux-5.10/drivers/tty/sysrq.c.66~	2020-12-13 23:41:30.000000000 +0100
+++ linux-5.10/drivers/tty/sysrq.c	2021-01-19 23:09:45.408605180 +0100
@@ -51,6 +51,7 @@
 #include <linux/syscalls.h>
 #include <linux/of.h>
 #include <linux/rcupdate.h>
+#include <linux/bootsplash.h>
 
 #include <asm/ptrace.h>
 #include <asm/irq_regs.h>
@@ -119,6 +120,7 @@ static void sysrq_handle_SAK(int key)
 {
 	struct work_struct *SAK_work = &vc_cons[fg_console].SAK_work;
 	schedule_work(SAK_work);
+	bootsplash_disable();
 }
 static const struct sysrq_key_op sysrq_SAK_op = {
 	.handler	= sysrq_handle_SAK,
