--- linux-4.13/arch/x86/entry/common.c.0250~	2017-09-03 22:56:17.000000000 +0200
+++ linux-4.13/arch/x86/entry/common.c	2017-09-04 14:48:56.658723194 +0200
@@ -178,7 +178,7 @@ static void exit_to_usermode_loop(struct
 }
 
 /* Called with IRQs disabled. */
-__visible inline void prepare_exit_to_usermode(struct pt_regs *regs)
+__visible extern inline void prepare_exit_to_usermode(struct pt_regs *regs)
 {
 	struct thread_info *ti = current_thread_info();
 	u32 cached_flags;
@@ -241,7 +241,7 @@ static void syscall_slow_exit_work(struc
  * Called with IRQs on and fully valid regs.  Returns with IRQs off in a
  * state such that we can immediately switch to user mode.
  */
-__visible inline void syscall_return_slowpath(struct pt_regs *regs)
+__visible extern inline void syscall_return_slowpath(struct pt_regs *regs)
 {
 	struct thread_info *ti = current_thread_info();
 	u32 cached_flags = READ_ONCE(ti->flags);
--- linux-4.13/drivers/staging/rtl8192u/Makefile.0250~	2017-09-03 22:56:17.000000000 +0200
+++ linux-4.13/drivers/staging/rtl8192u/Makefile	2017-09-04 14:48:56.658723194 +0200
@@ -1,6 +1,6 @@
 NIC_SELECT = RTL8192U
 
-ccflags-y := -std=gnu89
+ccflags-y := -std=gnu11
 ccflags-y += -O2
 
 ccflags-y += -DCONFIG_FORCE_HARD_FLOAT=y
--- linux-4.13/drivers/staging/rtl8723bs/include/ieee80211.h.0250~	2017-09-04 14:48:56.659723191 +0200
+++ linux-4.13/drivers/staging/rtl8723bs/include/ieee80211.h	2017-09-04 14:50:35.319423023 +0200
@@ -1008,18 +1008,18 @@ enum ieee80211_state {
 #define IP_FMT "%pI4"
 #define IP_ARG(x) (x)
 
-extern __inline int is_multicast_mac_addr(const u8 *addr)
+static inline int is_multicast_mac_addr(const u8 *addr)
 {
         return ((addr[0] != 0xff) && (0x01 & addr[0]));
 }
 
-extern __inline int is_broadcast_mac_addr(const u8 *addr)
+static inline int is_broadcast_mac_addr(const u8 *addr)
 {
 	return ((addr[0] == 0xff) && (addr[1] == 0xff) && (addr[2] == 0xff) &&   \
 		(addr[3] == 0xff) && (addr[4] == 0xff) && (addr[5] == 0xff));
 }
 
-extern __inline int is_zero_mac_addr(const u8 *addr)
+static inline int is_zero_mac_addr(const u8 *addr)
 {
 	return ((addr[0] == 0x00) && (addr[1] == 0x00) && (addr[2] == 0x00) &&   \
 		(addr[3] == 0x00) && (addr[4] == 0x00) && (addr[5] == 0x00));
--- linux-4.13/lib/decompress_unlzo.c.0250~	2017-09-03 22:56:17.000000000 +0200
+++ linux-4.13/lib/decompress_unlzo.c	2017-09-04 14:48:56.659723191 +0200
@@ -52,7 +52,7 @@ static const unsigned char lzop_magic[]
 #define HEADER_SIZE_MIN       (9 + 7     + 4 + 8     + 1       + 4)
 #define HEADER_SIZE_MAX       (9 + 7 + 1 + 8 + 8 + 4 + 1 + 255 + 4)
 
-STATIC inline long INIT parse_header(u8 *input, long *skip, long in_len)
+static inline long INIT parse_header(u8 *input, long *skip, long in_len)
 {
 	int l;
 	u8 *parse = input;
--- linux-4.13/Makefile.0250~	2017-09-04 14:48:56.659723191 +0200
+++ linux-4.13/Makefile	2017-09-04 14:51:00.831345374 +0200
@@ -301,7 +301,7 @@ HOST_LFS_LIBS := $(shell getconf LFS_LIB
 HOSTCC       = gcc
 HOSTCXX      = g++
 HOSTCFLAGS   := -Wall -Wmissing-prototypes -Wstrict-prototypes -O2 \
-		-fomit-frame-pointer -std=gnu89 $(HOST_LFS_CFLAGS)
+		-fomit-frame-pointer -std=gnu11 $(HOST_LFS_CFLAGS)
 HOSTCXXFLAGS := -O2 $(HOST_LFS_CFLAGS)
 HOSTLDFLAGS  := $(HOST_LFS_LDFLAGS)
 HOST_LOADLIBES := $(HOST_LFS_LIBS)
@@ -399,7 +399,7 @@ KBUILD_CFLAGS   := -Wall -Wundef -Wstric
 		   -fno-strict-aliasing -fno-common -fshort-wchar \
 		   -Werror-implicit-function-declaration \
 		   -Wno-format-security \
-		   -std=gnu89 $(call cc-option,-fno-PIE)
+		   -std=gnu11 $(call cc-option,-fno-PIE)
 
 
 KBUILD_AFLAGS_KERNEL :=
--- linux-4.13/tools/testing/selftests/sync/Makefile.0250~	2017-09-03 22:56:17.000000000 +0200
+++ linux-4.13/tools/testing/selftests/sync/Makefile	2017-09-04 14:48:56.659723191 +0200
@@ -1,4 +1,4 @@
-CFLAGS += -O2 -g -std=gnu89 -pthread -Wall -Wextra
+CFLAGS += -O2 -g -std=gnu11 -pthread -Wall -Wextra
 CFLAGS += -I../../../../usr/include/
 LDFLAGS += -pthread
 
--- linux-4.13/drivers/video/fbdev/i810/i810_dvt.c.omv~	2017-09-04 15:52:19.146422533 +0200
+++ linux-4.13/drivers/video/fbdev/i810/i810_dvt.c	2017-09-04 15:53:17.307284532 +0200
@@ -207,7 +207,7 @@ void round_off_xres(u32 *xres)
 		*xres = 1600;
 }
 
-inline void round_off_yres(u32 *xres, u32 *yres)
+void round_off_yres(u32 *xres, u32 *yres)
 {
 	*yres = (*xres * 3) >> 2;
 }
