diff -up linux-5.3/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c.omv~ linux-5.3/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c
--- linux-5.3/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c.omv~	2019-09-17 18:07:34.820865759 +0200
+++ linux-5.3/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm.c	2019-09-17 18:07:47.092348171 +0200
@@ -2107,6 +2107,7 @@ static int amdgpu_dm_backlight_get_brigh
 }
 
 static const struct backlight_ops amdgpu_dm_backlight_ops = {
+	.options = BL_CORE_SUSPENDRESUME,
 	.get_brightness = amdgpu_dm_backlight_get_brightness,
 	.update_status	= amdgpu_dm_backlight_update_status,
 };
