--- linux-4.12/fs/btrfs/zstd.c.omv~	2017-08-17 21:12:35.674612608 +0200
+++ linux-4.12/fs/btrfs/zstd.c	2017-08-17 21:13:29.883512195 +0200
@@ -260,13 +260,9 @@ out:
 	return ret;
 }
 
-static int zstd_decompress_bio(struct list_head *ws, struct compressed_bio *cb)
+static int zstd_decompress_bio(struct list_head *ws, struct page **pages_in, u64 disk_start, struct bio *orig_bio, size_t srclen)
 {
 	struct workspace *workspace = list_entry(ws, struct workspace, list);
-	struct page **pages_in = cb->compressed_pages;
-	u64 disk_start = cb->start;
-	struct bio *orig_bio = cb->orig_bio;
-	size_t srclen = cb->compressed_len;
 	ZSTD_DStream *stream;
 	int ret = 0;
 	unsigned long page_in_index = 0;
