diff -up linux-5.12-rc1/init/Kconfig.1~ linux-5.12-rc1/init/Kconfig
--- linux-5.12-rc1/init/Kconfig.1~	2021-03-01 01:05:19.000000000 +0100
+++ linux-5.12-rc1/init/Kconfig	2021-03-01 02:26:16.884986987 +0100
@@ -2218,10 +2218,10 @@ config MODULE_COMPRESS
 	bool "Compress modules on installation"
 	help
 
-	  Compresses kernel modules when 'make modules_install' is run; gzip or
-	  xz depending on "Compression algorithm" below.
+	  Compresses kernel modules when 'make modules_install' is run; gzip,
+	  xz or zstd depending on "Compression algorithm" below.
 
-	  module-init-tools MAY support gzip, and kmod MAY support gzip and xz.
+	  module-init-tools MAY support gzip, and kmod MAY support gzip, xz and zstd.
 
 	  Out-of-tree kernel modules installed using Kbuild will also be
 	  compressed upon installation.
@@ -2241,7 +2241,7 @@ choice
 	  This determines which sort of compression will be used during
 	  'make modules_install'.
 
-	  GZIP (default) and XZ are supported.
+	  GZIP (default), XZ and ZSTD are supported.
 
 config MODULE_COMPRESS_GZIP
 	bool "GZIP"
@@ -2249,6 +2249,9 @@ config MODULE_COMPRESS_GZIP
 config MODULE_COMPRESS_XZ
 	bool "XZ"
 
+config MODULE_COMPRESS_ZSTD
+	bool "ZSTD"
+
 endchoice
 
 config MODULE_ALLOW_MISSING_NAMESPACE_IMPORTS
diff -up linux-5.12-rc1/Makefile.1~ linux-5.12-rc1/Makefile
--- linux-5.12-rc1/Makefile.1~	2021-03-01 02:26:16.884986987 +0100
+++ linux-5.12-rc1/Makefile	2021-03-01 02:26:51.428776570 +0100
@@ -1075,6 +1075,9 @@ ifdef CONFIG_MODULE_COMPRESS
   ifdef CONFIG_MODULE_COMPRESS_XZ
     mod_compress_cmd = $(XZ) --lzma2=dict=2MiB -f
   endif # CONFIG_MODULE_COMPRESS_XZ
+  ifdef CONFIG_MODULE_COMPRESS_ZSTD
+    mod_compress_cmd = zstd -T0 -19 -q --rm -f
+  endif # CONFIG_MODULE_COMPRESS_ZSTD
 endif # CONFIG_MODULE_COMPRESS
 export mod_compress_cmd
 
