diff -up linux-5.5/tools/power/cpupower/utils/idle_monitor/cpuidle_sysfs.c.omv~ linux-5.5/tools/power/cpupower/utils/idle_monitor/cpuidle_sysfs.c
--- linux-5.5/tools/power/cpupower/utils/idle_monitor/cpuidle_sysfs.c.omv~	2020-02-13 17:52:30.247315834 +0100
+++ linux-5.5/tools/power/cpupower/utils/idle_monitor/cpuidle_sysfs.c	2020-02-13 17:52:36.175315446 +0100
@@ -13,6 +13,8 @@
 #include "helpers/helpers.h"
 #include "idle_monitor/cpupower-monitor.h"
 
+int cpu_count;
+
 #define CPUIDLE_STATES_MAX 10
 static cstate_t cpuidle_cstates[CPUIDLE_STATES_MAX];
 struct cpuidle_monitor cpuidle_sysfs_monitor;
diff -up linux-5.5/tools/power/cpupower/utils/idle_monitor/cpupower-monitor.h.omv~ linux-5.5/tools/power/cpupower/utils/idle_monitor/cpupower-monitor.h
--- linux-5.5/tools/power/cpupower/utils/idle_monitor/cpupower-monitor.h.omv~	2020-02-13 17:51:59.327317859 +0100
+++ linux-5.5/tools/power/cpupower/utils/idle_monitor/cpupower-monitor.h	2020-02-13 17:52:07.205317343 +0100
@@ -25,7 +25,7 @@
 #endif
 #define CSTATE_DESC_LEN 60
 
-int cpu_count;
+extern int cpu_count;
 
 /* Hard to define the right names ...: */
 enum power_range_e {
diff -up linux-5.5/tools/power/cpupower/utils/idle_monitor/amd_fam14h_idle.c.omv~ linux-5.5/tools/power/cpupower/utils/idle_monitor/amd_fam14h_idle.c
--- linux-5.5/tools/power/cpupower/utils/idle_monitor/amd_fam14h_idle.c.omv~	2020-02-13 21:38:49.575426637 +0100
+++ linux-5.5/tools/power/cpupower/utils/idle_monitor/amd_fam14h_idle.c	2020-02-13 21:38:55.711426235 +0100
@@ -82,7 +82,7 @@ static struct pci_access *pci_acc;
 static struct pci_dev *amd_fam14h_pci_dev;
 static int nbp1_entered;
 
-struct timespec start_time;
+static struct timespec start_time;
 static unsigned long long timediff;
 
 #ifdef DEBUG
diff -up linux-5.5/tools/power/cpupower/utils/idle_monitor/cpuidle_sysfs.c.omv~ linux-5.5/tools/power/cpupower/utils/idle_monitor/cpuidle_sysfs.c
--- linux-5.5/tools/power/cpupower/utils/idle_monitor/cpuidle_sysfs.c.omv~	2020-02-13 21:38:30.488427887 +0100
+++ linux-5.5/tools/power/cpupower/utils/idle_monitor/cpuidle_sysfs.c	2020-02-13 21:38:40.148427254 +0100
@@ -21,7 +21,7 @@ struct cpuidle_monitor cpuidle_sysfs_mon
 
 static unsigned long long **previous_count;
 static unsigned long long **current_count;
-struct timespec start_time;
+static struct timespec start_time;
 static unsigned long long timediff;
 
 static int cpuidle_get_count_percent(unsigned int id, double *percent,
