--- linux-4.15/drivers/media/dvb-frontends/si2168b.c.omv~	2018-02-13 13:01:39.417460412 +0100
+++ linux-4.15/drivers/media/dvb-frontends/si2168b.c	2018-02-13 13:01:42.512477570 +0100
@@ -12,7 +12,6 @@
 #include <linux/jiffies.h>
 #include <asm/div64.h>
 
-#include "dvb_frontend.h"
 #include "si2168b_priv.h"
 #include "silabs_tercab.h"
 
--- linux-4.15/drivers/media/dvb-frontends/si2168b.h.omv~	2018-02-13 13:02:20.033684040 +0100
+++ linux-4.15/drivers/media/dvb-frontends/si2168b.h	2018-02-13 13:03:47.063152873 +0100
@@ -20,7 +20,7 @@ struct dvb_stream_ids {
 
 #include <linux/kconfig.h>
 #include <linux/dvb/frontend.h>
-#include "dvb_frontend.h"
+#include <media/dvb_frontend.h>
 
 extern int _sitrace; /* module parameter: SiTRACE on/off */
 
--- linux-4.15/drivers/media/dvb-frontends/silg.c.omv~	2018-02-13 13:04:49.114479534 +0100
+++ linux-4.15/drivers/media/dvb-frontends/silg.c	2018-02-13 13:04:56.647518801 +0100
@@ -12,7 +12,7 @@
 #include <linux/string.h>
 #include <linux/slab.h>
 
-#include "dvb_frontend.h"
+#include <media/dvb_frontend.h>
 #include "silg.h"
 
 static int debug = 0;
--- linux-4.15/drivers/media/dvb-frontends/silg.h.omv~	2018-02-13 13:05:45.946773861 +0100
+++ linux-4.15/drivers/media/dvb-frontends/silg.h	2018-02-13 13:05:51.591802862 +0100
@@ -3,7 +3,7 @@
 
 #include <linux/kconfig.h>
 #include <linux/dvb/frontend.h>
-#include "dvb_frontend.h"
+#include <media/dvb_frontend.h>
 
 #include "lgdt3306a.h"
 #include "si2168b.h"
--- linux-4.15/drivers/media/tuners/silabs_tercab.c.omv~	2018-02-13 13:09:01.875759417 +0100
+++ linux-4.15/drivers/media/tuners/silabs_tercab.c	2018-02-13 13:09:44.377968293 +0100
@@ -7,8 +7,10 @@
  *
  */
 
-#include  <linux/delay.h>
+#include <linux/delay.h>
 #include <linux/videodev2.h>
+#include <linux/module.h>
+#include <media/dvb_frontend.h>
 #include "tuner-i2c.h"
 #include "silabs_tercab_priv.h"
 
@@ -3441,7 +3443,6 @@ MODULE_DESCRIPTION("Silicon Labs terrest
 MODULE_AUTHOR("Source code provided by Silicon Laboratories Inc.");
 MODULE_AUTHOR("Henning Garbers <hgarbers@pctvsystems.com>");
 MODULE_LICENSE("PROPRIETARY AND CONFIDENTIAL");
-MODULE_VERSION("2015-02-05");
 
 /*
  * Overrides for Emacs so that we follow Linus's tabbing style.
--- linux-4.15/drivers/media/tuners/silabs_tercab.h.omv~	2018-02-13 13:03:16.122987712 +0100
+++ linux-4.15/drivers/media/tuners/silabs_tercab.h	2018-02-13 13:03:18.102998329 +0100
@@ -20,7 +20,6 @@
 #define __SILABS_TERCAB_H__
 
 #include <linux/i2c.h>
-#include "dvb_frontend.h"
 
 #define HVR19x5_QAM_IF 4000
 #define HVR19x5_VSB_IF 3250
