--- linux-3.12/kernel/power/tuxonice_ui.c.omv~	2014-01-17 16:40:14.115685561 +0100
+++ linux-3.12/kernel/power/tuxonice_ui.c	2014-01-17 16:41:37.749016248 +0100
@@ -83,8 +83,8 @@ static char toi_wait_for_keypress(int ti
 void toi_early_boot_message(int message_detail, int default_answer,
 	char *warning_reason, ...)
 {
-#if defined(CONFIG_VT) || defined(CONFIG_SERIAL_CONSOLE)
 	unsigned long orig_state = get_toi_state(), continue_req = 0;
+#if defined(CONFIG_VT) || defined(CONFIG_SERIAL_CONSOLE)
 	unsigned long orig_loglevel = console_loglevel;
 	int can_ask = 1;
 #else
