diff -up linux-5.13/mm/vmalloc.c.30~ linux-5.13/mm/vmalloc.c
--- linux-5.13/mm/vmalloc.c.30~	2021-06-30 02:05:21.596002648 +0200
+++ linux-5.13/mm/vmalloc.c	2021-06-30 02:07:22.121226062 +0200
@@ -2388,6 +2388,7 @@ struct vm_struct *__get_vm_area_caller(u
 	return __get_vm_area_node(size, 1, PAGE_SHIFT, flags, start, end,
 				  NUMA_NO_NODE, GFP_KERNEL, caller);
 }
+EXPORT_SYMBOL(__get_vm_area_caller);
 
 /**
  * get_vm_area - reserve a contiguous kernel virtual area
