--- linux-3.12/drivers/media/usb/cx231xx/Kconfig.h930c-3~	2014-01-10 14:14:36.829489807 +0100
+++ linux-3.12/drivers/media/usb/cx231xx/Kconfig	2014-01-10 14:15:53.049490805 +0100
@@ -45,6 +45,7 @@ config VIDEO_CX231XX_DVB
 	select MEDIA_TUNER_XC5000 if MEDIA_SUBDRV_AUTOSELECT
 	select MEDIA_TUNER_TDA18271 if MEDIA_SUBDRV_AUTOSELECT
 	select DVB_MB86A20S if MEDIA_SUBDRV_AUTOSELECT
+	select DVB_SI2165 if MEDIA_SUBDRV_AUTOSELECT
 
 	---help---
 	  This adds support for DVB cards based on the
--- linux-3.12/drivers/media/usb/cx231xx/cx231xx-cards.c.h930c-3~	2014-01-10 14:14:36.822823140 +0100
+++ linux-3.12/drivers/media/usb/cx231xx/cx231xx-cards.c	2014-01-10 14:14:36.829489807 +0100
@@ -720,7 +720,7 @@ struct cx231xx_board cx231xx_boards[] =
 		.gpio_pin_status_mask = 0x4001000,
 		.tuner_i2c_master = 1,
 		.demod_i2c_master = 2,
-		.has_dvb = 0,
+		.has_dvb = 1,
 		.demod_addr = 0x0e,
 		.norm = V4L2_STD_PAL,
 
--- linux-3.12/drivers/media/usb/cx231xx/cx231xx-dvb.c.h930c-3~	2013-11-04 00:41:51.000000000 +0100
+++ linux-3.12/drivers/media/usb/cx231xx/cx231xx-dvb.c	2014-01-10 14:14:36.829489807 +0100
@@ -32,6 +32,7 @@
 #include "tda18271.h"
 #include "s5h1411.h"
 #include "lgdt3305.h"
+#include "si2165.h"
 #include "mb86a20s.h"
 
 MODULE_DESCRIPTION("driver for cx231xx based DVB cards");
@@ -695,6 +696,30 @@ static int dvb_init(struct cx231xx *dev)
 			result = -EINVAL;
 			goto out_free;
 		}
+
+		/* define general-purpose callback pointer */
+		dvb->frontend->callback = cx231xx_tuner_callback;
+
+		dvb_attach(tda18271_attach, dev->dvb->frontend,
+			   0x60, &dev->i2c_bus[dev->board.tuner_i2c_master].i2c_adap,
+			   &hcw_tda18271_config);
+		break;
+
+	case CX231XX_BOARD_HAUPPAUGE_EXETER_CT:
+
+		printk(KERN_INFO "%s: looking for tuner / demod on i2c bus: %d\n",
+		       __func__, i2c_adapter_id(&dev->i2c_bus[dev->board.tuner_i2c_master].i2c_adap));
+
+		dev->dvb->frontend = dvb_attach(si2165_attach,
+						0x64,
+						&dev->i2c_bus[dev->board.tuner_i2c_master].i2c_adap);
+
+		if (dev->dvb->frontend == NULL) {
+			printk(DRIVER_NAME
+			       ": Failed to attach SI2165 front end\n");
+			result = -EINVAL;
+			goto out_free;
+		}
 
 		/* define general-purpose callback pointer */
 		dvb->frontend->callback = cx231xx_tuner_callback;
