diff -up linux-5.8/drivers/net/wireless/intel/iwlwifi/pcie/drv.c.4~ linux-5.8/drivers/net/wireless/intel/iwlwifi/pcie/drv.c
--- linux-5.8/drivers/net/wireless/intel/iwlwifi/pcie/drv.c.4~	2020-09-02 16:25:03.480170474 +0200
+++ linux-5.8/drivers/net/wireless/intel/iwlwifi/pcie/drv.c	2020-09-02 16:26:33.440420919 +0200
@@ -588,9 +588,10 @@ static const struct iwl_dev_info iwl_dev
 	IWL_DEV_INFO(0x271C, 0x0214, iwl9260_2ac_cfg, iwl9260_1_name),
 
 /* AX200 */
+	/* IWL_CFG_ANY need to be the first entry, otherwise everything is overridden */
+	IWL_DEV_INFO(0x2723, IWL_CFG_ANY, iwl_ax200_cfg_cc, iwl_ax200_name),
 	IWL_DEV_INFO(0x2723, 0x1653, iwl_ax200_cfg_cc, iwl_ax200_killer_1650w_name),
 	IWL_DEV_INFO(0x2723, 0x1654, iwl_ax200_cfg_cc, iwl_ax200_killer_1650x_name),
-	IWL_DEV_INFO(0x2723, IWL_CFG_ANY, iwl_ax200_cfg_cc, iwl_ax200_name),
 
 	/* QnJ with Hr */
 	IWL_DEV_INFO(0x2720, IWL_CFG_ANY, iwl_qnj_b0_hr_b0_cfg, iwl_ax201_name),
