diff -up linux-5.11-rc1/drivers/hwmon/k10temp.c.omv~ linux-5.11-rc1/drivers/hwmon/k10temp.c
--- linux-5.11-rc1/drivers/hwmon/k10temp.c.omv~	2020-12-28 17:27:58.471181374 +0100
+++ linux-5.11-rc1/drivers/hwmon/k10temp.c	2020-12-28 17:28:25.797727529 +0100
@@ -460,7 +460,8 @@ static int k10temp_probe(struct pci_dev
 		data->is_zen = true;
 
 		switch (boot_cpu_data.x86_model) {
-		case 0x0 ... 0x1:	/* Zen3 */
+		case 0x0 ... 0x1:	/* Zen3 SP3/TR */
+		case 0x21:		/* ZEN3 Ryzen Desktop */
 			k10temp_get_ccd_support(pdev, data, 8);
 			break;
 		}
