--- linux-3.12/drivers/gpu/drm/i915/intel_display.c.i915~	2013-12-22 14:21:19.034482023 +0100
+++ linux-3.12/drivers/gpu/drm/i915/intel_display.c	2013-12-22 14:23:35.954483820 +0100
@@ -10083,6 +10083,9 @@ static struct intel_quirk intel_quirks[]
 	/* Acer Aspire 4736Z */
 	{ 0x2a42, 0x1025, 0x0260, quirk_invert_brightness },
 
+	/* Acer Aspire V3-772G */
+	{ 0x0416, 0x1025, 0x0781, quirk_invert_brightness },
+
 	/* Dell XPS13 HD Sandy Bridge */
 	{ 0x0116, 0x1028, 0x052e, quirk_no_pcm_pwm_enable },
 	/* Dell XPS13 HD and XPS13 FHD Ivy Bridge */
