diff -up linux-5.7-rc2/virt/kvm/kvm_main.c.omv~ linux-5.7-rc2/virt/kvm/kvm_main.c
--- linux-5.7-rc2/virt/kvm/kvm_main.c.omv~	2020-04-22 00:20:47.468689096 +0200
+++ linux-5.7-rc2/virt/kvm/kvm_main.c	2020-04-22 00:20:54.727162840 +0200
@@ -2425,7 +2425,7 @@ static int __kvm_gfn_to_hva_cache_init(s
 	gfn_t start_gfn = gpa >> PAGE_SHIFT;
 	gfn_t end_gfn = (gpa + len - 1) >> PAGE_SHIFT;
 	gfn_t nr_pages_needed = end_gfn - start_gfn + 1;
-	gfn_t nr_pages_avail;
+	gfn_t nr_pages_avail = 0;
 
 	/* Update ghc->generation before performing any error checks. */
 	ghc->generation = slots->generation;
