--- linux-4.9/drivers/gpu/drm/i915/intel_hdmi.c.0004~	2016-12-29 01:29:33.626367431 +0100
+++ linux-4.9/drivers/gpu/drm/i915/intel_hdmi.c	2016-12-29 01:31:46.622823104 +0100
@@ -1282,11 +1282,11 @@ intel_hdmi_mode_valid(struct drm_connect
 		clock *= 2;
 
 	/* check if we can do 8bpc */
-	status = hdmi_port_clock_valid(hdmi, clock, true);
+	status = hdmi_port_clock_valid(hdmi, clock, false);
 
 	/* if we can't do 8bpc we may still be able to do 12bpc */
 	if (!HAS_GMCH_DISPLAY(dev_priv) && status != MODE_OK)
-		status = hdmi_port_clock_valid(hdmi, clock * 3 / 2, true);
+		status = hdmi_port_clock_valid(hdmi, clock * 3 / 2, false);
 
 	return status;
 }
