diff -up linux-4.19/arch/arm64/kernel/vdso/Makefile.0005~ linux-4.19/arch/arm64/kernel/vdso/Makefile
--- linux-4.19/arch/arm64/kernel/vdso/Makefile.0005~	2018-10-22 08:37:37.000000000 +0200
+++ linux-4.19/arch/arm64/kernel/vdso/Makefile	2018-12-17 19:56:36.165686769 +0100
@@ -14,7 +14,7 @@ obj-vdso := $(addprefix $(obj)/, $(obj-v
 
 ccflags-y := -shared -fno-common -fno-builtin
 ccflags-y += -nostdlib -Wl,-soname=linux-vdso.so.1 \
-		$(call cc-ldoption, -Wl$(comma)--hash-style=sysv)
+		$(call cc-ldoption, -Wl$(comma)--hash-style=sysv -fuse-ld=bfd)
 
 # Disable gcov profiling for VDSO code
 GCOV_PROFILE := n
diff -up linux-4.19/arch/x86/entry/vdso/Makefile.0005~ linux-4.19/arch/x86/entry/vdso/Makefile
--- linux-4.19/arch/x86/entry/vdso/Makefile.0005~	2018-12-17 19:56:36.166686783 +0100
+++ linux-4.19/arch/x86/entry/vdso/Makefile	2018-12-17 19:58:18.883124493 +0100
@@ -47,7 +47,7 @@ targets += $(vdso_img_sodbg) $(vdso_img-
 CPPFLAGS_vdso.lds += -P -C
 
 VDSO_LDFLAGS_vdso.lds = -m elf_x86_64 -soname linux-vdso.so.1 --no-undefined \
-			-z max-page-size=4096
+			-z max-page-size=4096 -fuse-ld=bfd
 
 $(obj)/vdso64.so.dbg: $(obj)/vdso.lds $(vobjs) FORCE
 	$(call if_changed,vdso)
@@ -98,7 +98,7 @@ CFLAGS_REMOVE_vvar.o = -pg
 
 CPPFLAGS_vdsox32.lds = $(CPPFLAGS_vdso.lds)
 VDSO_LDFLAGS_vdsox32.lds = -m elf32_x86_64 -soname linux-vdso.so.1 \
-			   -z max-page-size=4096
+			   -z max-page-size=4096 -fuse-ld=bfd
 
 # x32-rebranded versions
 vobjx32s-y := $(vobjs-y:.o=-x32.o)
