diff -up linux-4.17/arch/arm64/kernel/vdso/Makefile.0005~ linux-4.17/arch/arm64/kernel/vdso/Makefile
--- linux-4.17/arch/arm64/kernel/vdso/Makefile.0005~	2018-06-03 23:15:21.000000000 +0200
+++ linux-4.17/arch/arm64/kernel/vdso/Makefile	2018-06-21 14:36:07.860889532 +0200
@@ -14,7 +14,7 @@ obj-vdso := $(addprefix $(obj)/, $(obj-v
 
 ccflags-y := -shared -fno-common -fno-builtin
 ccflags-y += -nostdlib -Wl,-soname=linux-vdso.so.1 \
-		$(call cc-ldoption, -Wl$(comma)--hash-style=sysv)
+		$(call cc-ldoption, -Wl$(comma)--hash-style=sysv -fuse-ld=bfd)
 
 # Disable gcov profiling for VDSO code
 GCOV_PROFILE := n
diff -up linux-4.17/arch/x86/entry/vdso/Makefile.0005~ linux-4.17/arch/x86/entry/vdso/Makefile
--- linux-4.17/arch/x86/entry/vdso/Makefile.0005~	2018-06-21 14:36:07.860889532 +0200
+++ linux-4.17/arch/x86/entry/vdso/Makefile	2018-06-21 14:37:46.237884111 +0200
@@ -49,6 +49,7 @@ CPPFLAGS_vdso.lds += -P -C
 VDSO_LDFLAGS_vdso.lds = -m64 -Wl,-soname=linux-vdso.so.1 \
 			-Wl,--no-undefined \
 			-Wl,-z,max-page-size=4096 -Wl,-z,common-page-size=4096 \
+			-fuse-ld=bfd \
 			$(DISABLE_LTO)
 
 $(obj)/vdso64.so.dbg: $(obj)/vdso.lds $(vobjs) FORCE
@@ -98,7 +99,8 @@ CPPFLAGS_vdsox32.lds = $(CPPFLAGS_vdso.l
 VDSO_LDFLAGS_vdsox32.lds = -Wl,-m,elf32_x86_64 \
 			   -Wl,-soname=linux-vdso.so.1 \
 			   -Wl,-z,max-page-size=4096 \
-			   -Wl,-z,common-page-size=4096
+			   -Wl,-z,common-page-size=4096 \
+			   -fuse-ld=bfd
 
 # x32-rebranded versions
 vobjx32s-y := $(vobjs-y:.o=-x32.o)
