diff -up linux-5.13-rc1/drivers/gpu/drm/i915/display/intel_hdmi.c.5~ linux-5.13-rc1/drivers/gpu/drm/i915/display/intel_hdmi.c
--- linux-5.13-rc1/drivers/gpu/drm/i915/display/intel_hdmi.c.5~	2021-05-11 00:57:41.046059747 +0200
+++ linux-5.13-rc1/drivers/gpu/drm/i915/display/intel_hdmi.c	2021-05-11 00:58:41.046025749 +0200
@@ -1893,13 +1893,13 @@ intel_hdmi_mode_valid(struct drm_connect
 
 	/* check if we can do 8bpc */
 	status = hdmi_port_clock_valid(hdmi, intel_hdmi_port_clock(clock, 8),
-				       true, has_hdmi_sink);
+				       false, has_hdmi_sink);
 
 	if (has_hdmi_sink) {
 		/* if we can't do 8bpc we may still be able to do 12bpc */
 		if (status != MODE_OK && !HAS_GMCH(dev_priv))
 			status = hdmi_port_clock_valid(hdmi, intel_hdmi_port_clock(clock, 12),
-						       true, has_hdmi_sink);
+						       false, has_hdmi_sink);
 
 		/* if we can't do 8,12bpc we may still be able to do 10bpc */
 		if (status != MODE_OK && DISPLAY_VER(dev_priv) >= 11)
