diff -up linux-5.1/drivers/gpu/drm/i915/intel_hdmi.c.3~ linux-5.1/drivers/gpu/drm/i915/intel_hdmi.c
--- linux-5.1/drivers/gpu/drm/i915/intel_hdmi.c.3~	2019-05-07 17:17:24.996151319 +0200
+++ linux-5.1/drivers/gpu/drm/i915/intel_hdmi.c	2019-05-07 17:19:03.873570987 +0200
@@ -1584,13 +1584,13 @@ intel_hdmi_mode_valid(struct drm_connect
 		clock /= 2;
 
 	/* check if we can do 8bpc */
-	status = hdmi_port_clock_valid(hdmi, clock, true, force_dvi);
+	status = hdmi_port_clock_valid(hdmi, clock, false, force_dvi);
 
 	if (hdmi->has_hdmi_sink && !force_dvi) {
 		/* if we can't do 8bpc we may still be able to do 12bpc */
 		if (status != MODE_OK && !HAS_GMCH(dev_priv))
 			status = hdmi_port_clock_valid(hdmi, clock * 3 / 2,
-						       true, force_dvi);
+						       false, force_dvi);
 
 		/* if we can't do 8,12bpc we may still be able to do 10bpc */
 		if (status != MODE_OK && INTEL_GEN(dev_priv) >= 11)
