diff -up linux-5.3-rc1/arch/arm64/kernel/vdso/Makefile.4~ linux-5.3-rc1/arch/arm64/kernel/vdso/Makefile
--- linux-5.3-rc1/arch/arm64/kernel/vdso/Makefile.4~	2019-07-24 00:26:39.079420412 +0200
+++ linux-5.3-rc1/arch/arm64/kernel/vdso/Makefile	2019-07-24 00:32:03.196644465 +0200
@@ -17,7 +17,7 @@ obj-vdso := vgettimeofday.o note.o sigre
 targets := $(obj-vdso) vdso.so vdso.so.dbg
 obj-vdso := $(addprefix $(obj)/, $(obj-vdso))
 
-ldflags-y := -shared -nostdlib -soname=linux-vdso.so.1 --hash-style=sysv \
+ldflags-y := -shared -nostdlib -soname=linux-vdso.so.1 --hash-style=sysv -fuse-ld=bfd \
 		--build-id -n -T
 
 ccflags-y := -fno-common -fno-builtin -fno-stack-protector -ffixed-x18
diff -up linux-5.3-rc1/arch/x86/entry/vdso/Makefile.4~ linux-5.3-rc1/arch/x86/entry/vdso/Makefile
--- linux-5.3-rc1/arch/x86/entry/vdso/Makefile.4~	2019-07-21 23:05:38.000000000 +0200
+++ linux-5.3-rc1/arch/x86/entry/vdso/Makefile	2019-07-24 00:32:22.289564153 +0200
@@ -53,7 +53,7 @@ targets += $(vdso_img_sodbg) $(vdso_img-
 CPPFLAGS_vdso.lds += -P -C
 
 VDSO_LDFLAGS_vdso.lds = -m elf_x86_64 -soname linux-vdso.so.1 --no-undefined \
-			-z max-page-size=4096
+			-z max-page-size=4096 -fuse-ld=bfd
 
 $(obj)/vdso64.so.dbg: $(obj)/vdso.lds $(vobjs) FORCE
 	$(call if_changed,vdso_and_check)
@@ -104,7 +104,7 @@ CFLAGS_REMOVE_vvar.o = -pg
 
 CPPFLAGS_vdsox32.lds = $(CPPFLAGS_vdso.lds)
 VDSO_LDFLAGS_vdsox32.lds = -m elf32_x86_64 -soname linux-vdso.so.1 \
-			   -z max-page-size=4096
+			   -z max-page-size=4096 -fuse-ld=bfd
 
 # x32-rebranded versions
 vobjx32s-y := $(vobjs-y:.o=-x32.o)
