diff -up linux-5.17/drivers/gpu/drm/i915/display/intel_hdmi.c.4~ linux-5.17/drivers/gpu/drm/i915/display/intel_hdmi.c
--- linux-5.17/drivers/gpu/drm/i915/display/intel_hdmi.c.4~	2022-03-26 17:10:30.108800943 +0100
+++ linux-5.17/drivers/gpu/drm/i915/display/intel_hdmi.c	2022-03-26 17:12:42.514427378 +0100
@@ -1939,14 +1939,14 @@ intel_hdmi_mode_clock_valid(struct drm_c
 
 	/* check if we can do 8bpc */
 	status = hdmi_port_clock_valid(hdmi, intel_hdmi_tmds_clock(clock, 8, ycbcr420_output),
-				       true, has_hdmi_sink);
+				       false, has_hdmi_sink);
 
 	/* if we can't do 8bpc we may still be able to do 12bpc */
 	if (status != MODE_OK &&
 	    intel_hdmi_source_bpc_possible(i915, 12) &&
 	    intel_hdmi_sink_bpc_possible(connector, 12, has_hdmi_sink, ycbcr420_output))
 		status = hdmi_port_clock_valid(hdmi, intel_hdmi_tmds_clock(clock, 12, ycbcr420_output),
-					       true, has_hdmi_sink);
+					       false, has_hdmi_sink);
 
 	/* if we can't do 8,12bpc we may still be able to do 10bpc */
 	if (status != MODE_OK &&
