diff -up linux-5.0/drivers/gpu/drm/i915/intel_guc_submission.c.omv~ linux-5.0/drivers/gpu/drm/i915/intel_guc_submission.c
diff -up linux-5.0/drivers/gpu/drm/i915/Makefile.omv~ linux-5.0/drivers/gpu/drm/i915/Makefile
--- linux-5.0/drivers/gpu/drm/i915/Makefile.omv~	2019-05-02 21:12:17.763467256 +0200
+++ linux-5.0/drivers/gpu/drm/i915/Makefile	2019-05-02 21:12:41.870160981 +0200
@@ -22,6 +22,7 @@ subdir-ccflags-y += $(call cc-disable-wa
 subdir-ccflags-y += $(call cc-disable-warning, sign-compare)
 subdir-ccflags-y += $(call cc-disable-warning, sometimes-uninitialized)
 subdir-ccflags-y += $(call cc-disable-warning, initializer-overrides)
+subdir-ccflags-y += $(call cc-disable-warning, address-of-packed-member)
 subdir-ccflags-$(CONFIG_DRM_I915_WERROR) += -Werror
 
 # Fine grained warnings disable
