diff -up linux-5.6/tools/bpf/bpftool/jit_disasm.c.omv~ linux-5.6/tools/bpf/bpftool/jit_disasm.c
diff -up linux-5.6/tools/build/feature/Makefile.omv~ linux-5.6/tools/build/feature/Makefile
--- linux-5.6/tools/build/feature/Makefile.omv~	2020-04-02 06:07:08.488570621 +0200
+++ linux-5.6/tools/build/feature/Makefile	2020-04-02 06:07:19.983631490 +0200
@@ -224,7 +224,7 @@ $(OUTPUT)test-libbfd.bin:
 	$(BUILD) -DPACKAGE='"perf"' -lbfd -ldl
 
 $(OUTPUT)test-disassembler-four-args.bin:
-	$(BUILD) -DPACKAGE='"perf"' -lbfd -lopcodes
+	$(BUILD) -DPACKAGE='"perf"' -lbfd -ldl -lopcodes
 
 $(OUTPUT)test-reallocarray.bin:
 	$(BUILD)
