diff -up linux-5.13/tools/include/linux/compiler-gcc.h.2~ linux-5.13/tools/include/linux/compiler-gcc.h
--- linux-5.13/tools/include/linux/compiler-gcc.h.2~	2021-06-28 00:21:11.000000000 +0200
+++ linux-5.13/tools/include/linux/compiler-gcc.h	2021-07-09 01:09:40.704907665 +0200
@@ -16,9 +16,9 @@
 # define __fallthrough __attribute__ ((fallthrough))
 #endif
 
-#if GCC_VERSION >= 40300
+#if __has_attribute(error)
 # define __compiletime_error(message) __attribute__((error(message)))
-#endif /* GCC_VERSION >= 40300 */
+#endif /* __has_attribute(error) */
 
 /* &a[0] degrades to a pointer: a different type from an array */
 #define __must_be_array(a)	BUILD_BUG_ON_ZERO(__same_type((a), &(a)[0]))
