diff -up linux-4.19/drivers/gpu/drm/i915/i915_gem_gtt.c.26~ linux-4.19/drivers/gpu/drm/i915/i915_gem_gtt.c
--- linux-4.19/drivers/gpu/drm/i915/i915_gem_gtt.c.26~	2018-12-02 07:43:03.801081390 +0100
+++ linux-4.19/drivers/gpu/drm/i915/i915_gem_gtt.c	2018-12-02 07:43:52.712137630 +0100
@@ -1769,7 +1769,7 @@ static void gen6_dump_ppgtt(struct i915_
 				continue;
 
 			seq_printf(m, "\t\t(%03d, %04d) %08llx: ",
-				   pde, pte,
+				   pde, pte, (long long)
 				   (pde * GEN6_PTES + pte) * I915_GTT_PAGE_SIZE);
 			for (i = 0; i < 4; i++) {
 				if (vaddr[pte + i] != scratch_pte)
