diff -up linux-5.2/kernel/sched/MuQSS.c.omv~ linux-5.2/kernel/sched/MuQSS.c
--- linux-5.2/kernel/sched/MuQSS.c.omv~	2019-07-31 14:33:11.842604878 +0200
+++ linux-5.2/kernel/sched/MuQSS.c	2019-07-31 14:33:23.726553288 +0200
@@ -4465,6 +4465,7 @@ int can_nice(const struct task_struct *p
 	return (nice_rlim <= task_rlimit(p, RLIMIT_NICE) ||
 		capable(CAP_SYS_NICE));
 }
+EXPORT_SYMBOL_GPL(can_nice);
 
 #ifdef __ARCH_WANT_SYS_NICE
 
