diff -up linux-4.19/drivers/gpu/drm/nouveau/nouveau_backlight.c.0130~ linux-4.19/drivers/gpu/drm/nouveau/nouveau_backlight.c
--- linux-4.19/drivers/gpu/drm/nouveau/nouveau_backlight.c.0130~	2018-11-15 03:00:24.796266238 +0100
+++ linux-4.19/drivers/gpu/drm/nouveau/nouveau_backlight.c	2018-11-15 03:01:13.071708875 +0100
@@ -253,6 +253,7 @@ nouveau_backlight_init(struct drm_connec
 	case NV_DEVICE_INFO_V0_FERMI:
 	case NV_DEVICE_INFO_V0_KEPLER:
 	case NV_DEVICE_INFO_V0_MAXWELL:
+	case NV_DEVICE_INFO_V0_PASCAL:
 		ret = nv50_backlight_init(nv_encoder, &props, &ops);
 		break;
 	default:
