diff -up linux-5.8-rc1/drivers/nvme/host/pci.c.36~ linux-5.8-rc1/drivers/nvme/host/pci.c
--- linux-5.8-rc1/drivers/nvme/host/pci.c.36~	2020-06-14 21:45:04.000000000 +0200
+++ linux-5.8-rc1/drivers/nvme/host/pci.c	2020-06-16 22:07:57.763065097 +0200
@@ -2580,8 +2580,11 @@ static void nvme_reset_work(struct work_
 	dev->ctrl.max_integrity_segments = 1;
 
 	result = nvme_init_identify(&dev->ctrl);
-	if (result)
+	if (result) {
+		dev_warn(dev->ctrl.device,
+			"nvme_init_identify() failed\n");
 		goto out;
+	}
 
 	if (dev->ctrl.oacs & NVME_CTRL_OACS_SEC_SUPP) {
 		if (!dev->ctrl.opal_dev)
@@ -2603,13 +2606,19 @@ static void nvme_reset_work(struct work_
 
 	if (dev->ctrl.hmpre) {
 		result = nvme_setup_host_mem(dev);
-		if (result < 0)
+		if (result < 0) {
+			dev_warn(dev->ctrl.device,
+				"hmpre - nvme_setup_host_mem() failed\n");
 			goto out;
+		}
 	}
 
 	result = nvme_setup_io_queues(dev);
-	if (result)
+	if (result) {
+		dev_warn(dev->ctrl.device,
+			"nvme_setup_io_queues() failed\n");
 		goto out;
+	}
 
 	/*
 	 * Keep the controller around but remove all namespaces if we don't have
