diff -up linux-5.14-rc1/MAINTAINERS.79~ linux-5.14-rc1/MAINTAINERS
--- linux-5.14-rc1/MAINTAINERS.79~	2021-07-12 12:12:57.767687483 +0200
+++ linux-5.14-rc1/MAINTAINERS	2021-07-12 12:13:28.381892324 +0200
@@ -13245,6 +13245,13 @@ T:	git git://git.kernel.org/pub/scm/linu
 F:	Documentation/filesystems/ntfs.rst
 F:	fs/ntfs/
 
+NTFS3 FILESYSTEM
+M:	Konstantin Komarov <almaz.alexandrovich@paragon-software.com>
+S:	Supported
+W:	http://www.paragon-software.com/
+F:	Documentation/filesystems/ntfs3.rst
+F:	fs/ntfs3/
+
 NUBUS SUBSYSTEM
 M:	Finn Thain <fthain@linux-m68k.org>
 L:	linux-m68k@lists.linux-m68k.org
