diff --git a/drivers/tty/vt/keyboard.c b/drivers/tty/vt/keyboard.c
index f4166263bb3a..177b35182b7c 100644
--- a/drivers/tty/vt/keyboard.c
+++ b/drivers/tty/vt/keyboard.c
@@ -47,6 +47,8 @@
 
 #include <asm/irq_regs.h>
 
+#include <linux/bootsplash.h>
+
 extern void ctrl_alt_del(void);
 
 /*
@@ -1353,6 +1355,15 @@ static void kbd_keycode(unsigned int keycode, int down, int hw_raw)
 	}
 #endif
 
+	/* Deactivate bootsplash on ESC or Alt+Fxx VT switch */
+	if (((keycode >= KEY_F1 && keycode <= KEY_F12) || keycode == KEY_ESC)
+	    && bootsplash_is_enabled()) {
+		bootsplash_disable();
+
+		if (keycode == KEY_ESC)
+			return;
+	}
+
 	if (kbd->kbdmode == VC_MEDIUMRAW) {
 		/*
 		 * This is extended medium raw mode, with keys above 127
