diff -up linux-5.17/tools/tracing/rtla/Makefile.omv~ linux-5.17/tools/tracing/rtla/Makefile
--- linux-5.17/tools/tracing/rtla/Makefile.omv~	2022-03-26 21:48:35.505914845 +0100
+++ linux-5.17/tools/tracing/rtla/Makefile	2022-03-26 21:48:42.262956229 +0100
@@ -81,8 +81,6 @@ clean: doc_clean
 	@test ! -f rtla || rm rtla
 	@test ! -f rtla-static || rm rtla-static
 	@test ! -f src/rtla.o || rm src/rtla.o
-	@test ! -f $(TARBALL) || rm -f $(TARBALL)
-	@rm -rf *~ $(OBJ) *.tar.$(CEXT)
 
 tarball: clean
 	rm -rf $(NAME)-$(VERSION) && mkdir $(NAME)-$(VERSION)
diff -up linux-5.17/tools/tracing/rtla/.omv~ linux-5.17/tools/tracing/rtla/
