diff -up linux-4.18/drivers/media/dvb-frontends/Kconfig.0145~ linux-4.18/drivers/media/dvb-frontends/Kconfig
--- linux-4.18/drivers/media/dvb-frontends/Kconfig.0145~	2018-08-27 10:37:27.224119363 +0200
+++ linux-4.18/drivers/media/dvb-frontends/Kconfig	2018-08-27 10:37:27.353119834 +0200
@@ -822,6 +822,13 @@ config DVB_ISL6421
 	help
 	  An SEC control chip.
 
+config DVB_ISL6422
+	tristate "ISL6422[B] SEC controller"
+	depends on DVB_CORE && I2C
+	default m if !MEDIA_SUBDRV_AUTOSELECT
+	help
+	  A Dual SEC controller chip from Intersil
+
 config DVB_ISL6423
 	tristate "ISL6423 SEC controller"
 	depends on DVB_CORE && I2C
@@ -888,6 +895,21 @@ config DVB_AF9033
 	select REGMAP_I2C
 	default m if !MEDIA_SUBDRV_AUTOSELECT
 
+config DVB_TAS2101
+	tristate "Tmax TAS2101 based"
+	depends on DVB_CORE && I2C
+	default m if !MEDIA_SUBDRV_AUTOSELECT
+	help
+	  A DVB-S demod and tuner module.
+
+config DVB_STV0910
+	tristate "STV0910 based"
+	depends on DVB_CORE && I2C
+	default m if !MEDIA_SUBDRV_AUTOSELECT
+	help
+	  DVB-S/S2/DSS Multistandard Professional/Broadcast demodulators.
+	  Say Y when you want to support these frontends.
+
 config DVB_HORUS3A
 	tristate "Sony Horus3A tuner"
 	depends on DVB_CORE && I2C
diff -up linux-4.18/drivers/media/dvb-frontends/Makefile.0145~ linux-4.18/drivers/media/dvb-frontends/Makefile
--- linux-4.18/drivers/media/dvb-frontends/Makefile.0145~	2018-08-27 10:37:27.224119363 +0200
+++ linux-4.18/drivers/media/dvb-frontends/Makefile	2018-08-27 10:38:09.719272450 +0200
@@ -97,6 +97,7 @@ obj-$(CONFIG_DVB_STV6110x) += stv6110x.o
 obj-$(CONFIG_DVB_M88DS3103) += m88ds3103.o
 obj-$(CONFIG_DVB_MN88472) += mn88472.o
 obj-$(CONFIG_DVB_MN88473) += mn88473.o
+obj-$(CONFIG_DVB_ISL6422) += isl6422.o
 obj-$(CONFIG_DVB_ISL6423) += isl6423.o
 obj-$(CONFIG_DVB_EC100) += ec100.o
 obj-$(CONFIG_DVB_DS3000) += ds3000.o
@@ -124,6 +125,7 @@ obj-$(CONFIG_DVB_M88RS2000) += m88rs2000
 obj-$(CONFIG_DVB_AF9033) += af9033.o
 obj-$(CONFIG_DVB_AS102_FE) += as102_fe.o
 obj-$(CONFIG_DVB_GP8PSK_FE) += gp8psk-fe.o
+obj-$(CONFIG_DVB_TAS2101) += tas2101.o
 obj-$(CONFIG_DVB_TC90522) += tc90522.o
 obj-$(CONFIG_DVB_MN88443X) += mn88443x.o
 obj-$(CONFIG_DVB_HORUS3A) += horus3a.o
diff -up linux-4.18/drivers/media/pci/saa716x/Makefile.0145~ linux-4.18/drivers/media/pci/saa716x/Makefile
--- linux-4.18/drivers/media/pci/saa716x/Makefile.0145~	2018-04-03 19:54:42.764725113 +0200
+++ linux-4.18/drivers/media/pci/saa716x/Makefile	2018-08-27 10:37:27.353119834 +0200
@@ -27,4 +27,4 @@ obj-$(CONFIG_DVB_SAA716X_BUDGET)  += saa
 obj-$(CONFIG_DVB_SAA716X_HYBRID)  += saa716x_hybrid.o
 obj-$(CONFIG_DVB_SAA716X_FF)	  += saa716x_ff.o
 
-EXTRA_CFLAGS = -Idrivers/media/dvb-core/ -Idrivers/media/dvb-frontends/ -Idrivers/media/tuners/
+EXTRA_CFLAGS = -Idrivers/media/dvb-core/ -Idrivers/media/dvb-frontends/ -Idrivers/media/tuners/ -Idrivers/media/pci/ttpci
diff -up linux-4.18/drivers/media/pci/saa716x/saa716x_pci.c.0145~ linux-4.18/drivers/media/pci/saa716x/saa716x_pci.c
--- linux-4.18/drivers/media/pci/saa716x/saa716x_pci.c.0145~	2018-04-03 19:54:42.766725124 +0200
+++ linux-4.18/drivers/media/pci/saa716x/saa716x_pci.c	2018-08-27 10:37:27.353119834 +0200
@@ -1,3 +1,4 @@
+#include <linux/spinlock.h>
 #include <asm/io.h>
 #include <asm/pgtable.h>
 #include <asm/page.h>
diff -up linux-4.18/drivers/media/tuners/Kconfig.0145~ linux-4.18/drivers/media/tuners/Kconfig
--- linux-4.18/drivers/media/tuners/Kconfig.0145~	2018-08-12 22:41:04.000000000 +0200
+++ linux-4.18/drivers/media/tuners/Kconfig	2018-08-27 10:37:27.353119834 +0200
@@ -291,4 +291,18 @@ config MEDIA_TUNER_QM1D1B0004
 	default m if !MEDIA_SUBDRV_AUTOSELECT
 	help
 	  Sharp QM1D1B0004 ISDB-S tuner driver.
+
+config MEDIA_TUNER_AV201X
+	tristate "Airoha Technology AV201x silicon tuner"
+	depends on MEDIA_SUPPORT && I2C
+	default m if !MEDIA_SUBDRV_AUTOSELECT
+	help
+	  Airoha Technology AV201x silicon tuner driver.
+
+config MEDIA_TUNER_STV6120
+	tristate "ST Microelectronics STV6120 dual silicon tuner"
+	depends on MEDIA_SUPPORT && I2C
+	default m if !MEDIA_SUBDRV_AUTOSELECT
+	help
+	  ST Microelectronics STV6120 dual silicon tuner driver.
 endmenu
diff -up linux-4.18/drivers/media/tuners/Makefile.0145~ linux-4.18/drivers/media/tuners/Makefile
--- linux-4.18/drivers/media/tuners/Makefile.0145~	2018-08-12 22:41:04.000000000 +0200
+++ linux-4.18/drivers/media/tuners/Makefile	2018-08-27 10:37:27.353119834 +0200
@@ -44,5 +44,7 @@ obj-$(CONFIG_MEDIA_TUNER_QM1D1C0042) +=
 obj-$(CONFIG_MEDIA_TUNER_QM1D1B0004) += qm1d1b0004.o
 obj-$(CONFIG_MEDIA_TUNER_M88RS6000T) += m88rs6000t.o
 obj-$(CONFIG_MEDIA_TUNER_TDA18250) += tda18250.o
+obj-$(CONFIG_MEDIA_TUNER_AV201X) += av201x.o
+obj-$(CONFIG_MEDIA_TUNER_STV6120) += stv6120.o
 
 ccflags-y += -I$(srctree)/drivers/media/dvb-frontends
diff -up linux-4.18/drivers/media/tuners/tda18212.c.0145~ linux-4.18/drivers/media/tuners/tda18212.c
--- linux-4.18/drivers/media/tuners/tda18212.c.0145~	2018-08-27 10:37:27.348119816 +0200
+++ linux-4.18/drivers/media/tuners/tda18212.c	2018-08-27 10:37:27.353119834 +0200
@@ -153,7 +153,7 @@ static int tda18212_set_params(struct dv
 
 	buf[0] = 0x02;
 	buf[1] = bw_params[i][1];
-	buf[2] = priv->cfg->xtout ? 0x43 : 0x40; /* 0x03; default value */
+	buf[2] = dev->cfg.xtout ? 0x43 : 0x40; /* 0x03; default value */
 	buf[3] = DIV_ROUND_CLOSEST(if_khz, 50);
 	buf[4] = ((c->frequency / 1000) >> 16) & 0xff;
 	buf[5] = ((c->frequency / 1000) >>  8) & 0xff;
