diff -up linux-5.18-rc3/drivers/media/dvb-frontends/Kconfig.21~ linux-5.18-rc3/drivers/media/dvb-frontends/Kconfig
--- linux-5.18-rc3/drivers/media/dvb-frontends/Kconfig.21~	2022-04-17 22:57:31.000000000 +0200
+++ linux-5.18-rc3/drivers/media/dvb-frontends/Kconfig	2022-04-19 23:37:12.987473937 +0200
@@ -816,6 +816,21 @@ config DVB_HELENE
 	help
 	  Say Y when you want to support this frontend.
 
+config DVB_TAS2101
+	tristate "Tmax TAS2101 based"
+	depends on DVB_CORE && I2C
+	default m if !MEDIA_SUBDRV_AUTOSELECT
+	help
+	  A DVB-S demod and tuner module.
+
+config DVB_STV0910
+	tristate "STV0910 based"
+	depends on DVB_CORE && I2C
+	default m if !MEDIA_SUBDRV_AUTOSELECT
+	help
+	  DVB-S/S2/DSS Multistandard Professional/Broadcast demodulators.
+	  Say Y when you want to support these frontends.
+
 config DVB_HORUS3A
 	tristate "Sony Horus3A tuner"
 	depends on DVB_CORE && I2C
@@ -837,6 +852,13 @@ config DVB_ISL6421
 	help
 	  An SEC control chip.
 
+config DVB_ISL6422
+	tristate "ISL6422[B] SEC controller"
+	depends on DVB_CORE && I2C
+	default m if !MEDIA_SUBDRV_AUTOSELECT
+	help
+	  A Dual SEC controller chip from Intersil
+
 config DVB_ISL6423
 	tristate "ISL6423 SEC controller"
 	depends on DVB_CORE && I2C
diff -up linux-5.18-rc3/drivers/media/dvb-frontends/Makefile.21~ linux-5.18-rc3/drivers/media/dvb-frontends/Makefile
--- linux-5.18-rc3/drivers/media/dvb-frontends/Makefile.21~	2022-04-19 23:36:13.713862995 +0200
+++ linux-5.18-rc3/drivers/media/dvb-frontends/Makefile	2022-04-19 23:37:45.980965780 +0200
@@ -53,6 +53,7 @@ obj-$(CONFIG_DVB_DS3000) += ds3000.o
 obj-$(CONFIG_DVB_DUMMY_FE) += dvb_dummy_fe.o
 obj-$(CONFIG_DVB_EC100) += ec100.o
 obj-$(CONFIG_DVB_GP8PSK_FE) += gp8psk-fe.o
+obj-$(CONFIG_DVB_TAS2101) += tas2101.o
 obj-$(CONFIG_DVB_HELENE) += helene.o
 obj-$(CONFIG_DVB_HORUS3A) += horus3a.o
 obj-$(CONFIG_DVB_ISL6405) += isl6405.o
@@ -77,6 +78,7 @@ obj-$(CONFIG_DVB_MB86A20S) += mb86a20s.o
 obj-$(CONFIG_DVB_MN88443X) += mn88443x.o
 obj-$(CONFIG_DVB_MN88472) += mn88472.o
 obj-$(CONFIG_DVB_MN88473) += mn88473.o
+obj-$(CONFIG_DVB_ISL6422) += isl6422.o
 obj-$(CONFIG_DVB_MT312) += mt312.o
 obj-$(CONFIG_DVB_MT352) += mt352.o
 obj-$(CONFIG_DVB_MXL5XX) += mxl5xx.o
diff -up linux-5.18-rc3/drivers/media/pci/saa716x/Makefile.21~ linux-5.18-rc3/drivers/media/pci/saa716x/Makefile
--- linux-5.18-rc3/drivers/media/pci/saa716x/Makefile.21~	2018-04-03 19:54:42.764725113 +0200
+++ linux-5.18-rc3/drivers/media/pci/saa716x/Makefile	2022-04-19 23:36:13.713862995 +0200
@@ -27,4 +27,4 @@ obj-$(CONFIG_DVB_SAA716X_BUDGET)  += saa
 obj-$(CONFIG_DVB_SAA716X_HYBRID)  += saa716x_hybrid.o
 obj-$(CONFIG_DVB_SAA716X_FF)	  += saa716x_ff.o
 
-EXTRA_CFLAGS = -Idrivers/media/dvb-core/ -Idrivers/media/dvb-frontends/ -Idrivers/media/tuners/
+EXTRA_CFLAGS = -Idrivers/media/dvb-core/ -Idrivers/media/dvb-frontends/ -Idrivers/media/tuners/ -Idrivers/media/pci/ttpci
diff -up linux-5.18-rc3/drivers/media/pci/saa716x/saa716x_pci.c.21~ linux-5.18-rc3/drivers/media/pci/saa716x/saa716x_pci.c
--- linux-5.18-rc3/drivers/media/pci/saa716x/saa716x_pci.c.21~	2018-04-03 19:54:42.766725124 +0200
+++ linux-5.18-rc3/drivers/media/pci/saa716x/saa716x_pci.c	2022-04-19 23:36:13.717196344 +0200
@@ -1,3 +1,4 @@
+#include <linux/spinlock.h>
 #include <asm/io.h>
 #include <asm/pgtable.h>
 #include <asm/page.h>
diff -up linux-5.18-rc3/drivers/media/tuners/Kconfig.21~ linux-5.18-rc3/drivers/media/tuners/Kconfig
--- linux-5.18-rc3/drivers/media/tuners/Kconfig.21~	2022-04-19 23:36:13.717196344 +0200
+++ linux-5.18-rc3/drivers/media/tuners/Kconfig	2022-04-19 23:38:16.997783856 +0200
@@ -297,4 +297,17 @@ config MEDIA_TUNER_XC5000
 	  This device is only used inside a SiP called together with a
 	  demodulator for now.
 
+config MEDIA_TUNER_AV201X
+	tristate "Airoha Technology AV201x silicon tuner"
+	depends on MEDIA_SUPPORT && I2C
+	default m if !MEDIA_SUBDRV_AUTOSELECT
+	help
+	  Airoha Technology AV201x silicon tuner driver.
+
+config MEDIA_TUNER_STV6120
+	tristate "ST Microelectronics STV6120 dual silicon tuner"
+	depends on MEDIA_SUPPORT && I2C
+	default m if !MEDIA_SUBDRV_AUTOSELECT
+	help
+	  ST Microelectronics STV6120 dual silicon tuner driver.
 endmenu
diff -up linux-5.18-rc3/drivers/media/tuners/Makefile.21~ linux-5.18-rc3/drivers/media/tuners/Makefile
--- linux-5.18-rc3/drivers/media/tuners/Makefile.21~	2022-04-19 23:36:13.717196344 +0200
+++ linux-5.18-rc3/drivers/media/tuners/Makefile	2022-04-19 23:38:41.101236366 +0200
@@ -37,6 +37,8 @@ obj-$(CONFIG_MEDIA_TUNER_TDA18212) += td
 obj-$(CONFIG_MEDIA_TUNER_TDA18218) += tda18218.o
 obj-$(CONFIG_MEDIA_TUNER_TDA18250) += tda18250.o
 obj-$(CONFIG_MEDIA_TUNER_TDA18271) += tda18271.o
+obj-$(CONFIG_MEDIA_TUNER_AV201X) += av201x.o
+obj-$(CONFIG_MEDIA_TUNER_STV6120) += stv6120.o
 obj-$(CONFIG_MEDIA_TUNER_TDA827X) += tda827x.o
 obj-$(CONFIG_MEDIA_TUNER_TDA8290) += tda8290.o
 obj-$(CONFIG_MEDIA_TUNER_TDA9887) += tda9887.o
diff -up linux-5.18-rc3/drivers/media/tuners/tda18212.c.21~ linux-5.18-rc3/drivers/media/tuners/tda18212.c
--- linux-5.18-rc3/drivers/media/tuners/tda18212.c.21~	2022-04-19 23:36:13.707196298 +0200
+++ linux-5.18-rc3/drivers/media/tuners/tda18212.c	2022-04-19 23:36:13.717196344 +0200
@@ -140,7 +140,7 @@ static int tda18212_set_params(struct dv
 
 	buf[0] = 0x02;
 	buf[1] = bw_params[i][1];
-	buf[2] = priv->cfg->xtout ? 0x43 : 0x40; /* 0x03; default value */
+	buf[2] = dev->cfg.xtout ? 0x43 : 0x40; /* 0x03; default value */
 	buf[3] = DIV_ROUND_CLOSEST(if_khz, 50);
 	buf[4] = ((c->frequency / 1000) >> 16) & 0xff;
 	buf[5] = ((c->frequency / 1000) >>  8) & 0xff;
