diff -up linux-5.4/drivers/media/dvb-frontends/tas2101.c.omv~ linux-5.4/drivers/media/dvb-frontends/tas2101.c
--- linux-5.4/drivers/media/dvb-frontends/tas2101.c.omv~	2019-12-01 19:47:58.264500593 +0100
+++ linux-5.4/drivers/media/dvb-frontends/tas2101.c	2019-12-01 19:48:10.986546815 +0100
@@ -844,7 +844,7 @@ static int tas2101_tune(struct dvb_front
 	return tas2101_read_status(fe, status);
 }
 
-static int tas2101_get_algo(struct dvb_frontend *fe)
+static enum dvbfe_algo tas2101_get_algo(struct dvb_frontend *fe)
 {
 	return DVBFE_ALGO_HW;
 }
