diff -up linux-4.18/drivers/media/dvb-frontends/tas2101.c.omv~ linux-4.18/drivers/media/dvb-frontends/tas2101.c
--- linux-4.18/drivers/media/dvb-frontends/tas2101.c.omv~	2018-08-30 02:18:47.235690694 +0200
+++ linux-4.18/drivers/media/dvb-frontends/tas2101.c	2018-08-30 02:19:07.014801765 +0200
@@ -868,10 +868,10 @@ static struct dvb_frontend_ops tas2101_o
 	.delsys = { SYS_DVBS, SYS_DVBS2 },
 	.info = {
 		.name = "Tmax TAS2101",
-		.frequency_min = 950000,
-		.frequency_max = 2150000,
-		.frequency_stepsize = 1011, /* kHz for QPSK frontends */
-		.frequency_tolerance = 5000,
+		.frequency_min_hz = 950000,
+		.frequency_max_hz = 2150000,
+		.frequency_stepsize_hz = 1011, /* kHz for QPSK frontends */
+		.frequency_tolerance_hz = 5000,
 		.symbol_rate_min = 1000000,
 		.symbol_rate_max = 45000000,
 		.caps = FE_CAN_INVERSION_AUTO |
diff -up linux-4.18/drivers/media/dvb-frontends/tas2101_priv.h.omv~ linux-4.18/drivers/media/dvb-frontends/tas2101_priv.h
--- linux-4.18/drivers/media/dvb-frontends/tas2101_priv.h.omv~	2018-08-30 02:18:22.803553489 +0200
+++ linux-4.18/drivers/media/dvb-frontends/tas2101_priv.h	2018-08-30 02:18:42.718665327 +0200
@@ -21,6 +21,8 @@
 #ifndef TAS2101_PRIV_H
 #define TAS2101_PRIV_H
 
+#include <linux/version.h>
+
 struct tas2101_priv {
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 7, 0)
 	struct i2c_mux_core *muxc;
diff -up linux-4.18/drivers/media/pci/saa716x/saa716x_ff_main.c.omv~ linux-4.18/drivers/media/pci/saa716x/saa716x_ff_main.c
--- linux-4.18/drivers/media/pci/saa716x/saa716x_ff_main.c.omv~	2018-08-30 02:28:28.459947244 +0200
+++ linux-4.18/drivers/media/pci/saa716x/saa716x_ff_main.c	2018-08-30 02:28:40.223973722 +0200
@@ -427,11 +427,6 @@ static int do_dvb_audio_ioctl(struct dvb
 	int ret = 0;
 
 	switch (cmd) {
-	case AUDIO_GET_PTS:
-	{
-		*(u64 *)parg = sti7109->audio_pts;
-		break;
-	}
 	default:
 		ret = -ENOIOCTLCMD;
 		break;
diff -up linux-4.18/drivers/media/pci/saa716x/saa716x_pci.c.omv~ linux-4.18/drivers/media/pci/saa716x/saa716x_pci.c
--- linux-4.18/drivers/media/pci/saa716x/saa716x_pci.c.omv~	2018-08-30 02:26:29.498720354 +0200
+++ linux-4.18/drivers/media/pci/saa716x/saa716x_pci.c	2018-08-30 02:26:48.183750694 +0200
@@ -7,6 +7,7 @@
 #include <linux/init.h>
 #include <linux/device.h>
 #include <linux/delay.h>
+#include <linux/version.h>
 
 #include <linux/signal.h>
 #include <linux/sched.h>
diff -up linux-4.18/drivers/media/tuners/av201x.c.omv~ linux-4.18/drivers/media/tuners/av201x.c
--- linux-4.18/drivers/media/tuners/av201x.c.omv~	2018-08-30 02:31:32.614430339 +0200
+++ linux-4.18/drivers/media/tuners/av201x.c	2018-08-30 02:31:43.976464405 +0200
@@ -262,9 +262,9 @@ static const struct dvb_tuner_ops av201x
 	.info = {
 		.name           = "Airoha Technology AV201x",
 
-		.frequency_min  = 850000,
-		.frequency_max  = 2300000,
-		.frequency_step = 206,
+		.frequency_min_hz  = 850000,
+		.frequency_max_hz  = 2300000,
+		.frequency_step_hz = 206,
 	},
 
 	.release = av201x_release,
diff -up linux-4.18/drivers/media/tuners/stv6120.c.omv~ linux-4.18/drivers/media/tuners/stv6120.c
--- linux-4.18/drivers/media/tuners/stv6120.c.omv~	2018-08-30 02:32:33.325617258 +0200
+++ linux-4.18/drivers/media/tuners/stv6120.c	2018-08-30 02:32:40.600640433 +0200
@@ -539,9 +539,9 @@ static int get_rf_strength(struct dvb_fr
 static struct dvb_tuner_ops tuner_ops = {
 	.info = {
 		.name = "STV6120",
-		.frequency_min  =  950000,
-		.frequency_max  = 2150000,
-		.frequency_step =       0
+		.frequency_min_hz  =  950000,
+		.frequency_max_hz  = 2150000,
+		.frequency_step_hz =       0
 	},
 	.init              = init,
 	.sleep             = sleep,
