diff -up linux-5.16/drivers/nvme/host/core.c.12~ linux-5.16/drivers/nvme/host/core.c
--- linux-5.16/drivers/nvme/host/core.c.12~	2022-01-10 11:23:40.255182494 +0100
+++ linux-5.16/drivers/nvme/host/core.c	2022-01-10 11:25:22.607566090 +0100
@@ -2504,7 +2504,16 @@ static const struct nvme_core_quirk_entr
 		.vid = 0x1e0f,
 		.mn = "KCD6XVUL6T40",
 		.quirks = NVME_QUIRK_NO_APST,
-	}
+	},
+        {
+		/* https://forum.openmandriva.org/t/nvme-ssd-m2-not-seen-by-omlx-4-0/2407
+		 * Unknown SSD ... Maybe ADATA/Hynix (a similar mn from ADTA
+		 * but vid seems to be Hynix)
+		 */
+		.vid = 0x1c5c,
+		.mn = "HFM128GDHTNG-8310B",
+		.quirks = NVME_QUIRK_NO_APST,
+ 	}
 };
 
 /* match is null-terminated but idstr is space-padded. */
