diff -up linux-5.4/drivers/nvme/host/core.c.35~ linux-5.4/drivers/nvme/host/core.c
--- linux-5.4/drivers/nvme/host/core.c.35~	2019-12-18 20:23:29.167152525 +0100
+++ linux-5.4/drivers/nvme/host/core.c	2019-12-18 20:34:05.573121649 +0100
@@ -2406,6 +2406,15 @@ static const struct nvme_core_quirk_entr
 		.vid = 0x14a4,
 		.fr = "22301111",
 		.quirks = NVME_QUIRK_SIMPLE_SUSPEND,
+	},
+        {
+		/* https://forum.openmandriva.org/t/nvme-ssd-m2-not-seen-by-omlx-4-0/2407
+		 * Unknown SSD ... Maybe ADATA/Hynix (a similar mn from ADTA
+		 * but vid seems to be Hynix)
+		 */
+		.vid = 0x1c5c,
+		.mn = "HFM128GDHTNG-8310B",
+		.quirks = NVME_QUIRK_NO_APST,
 	}
 };
 
