diff -up linux-5.2-rc1/arch/x86/boot/compressed/Makefile.20~ linux-5.2-rc1/arch/x86/boot/compressed/Makefile
--- linux-5.2-rc1/arch/x86/boot/compressed/Makefile.20~	2019-05-20 00:47:09.000000000 +0200
+++ linux-5.2-rc1/arch/x86/boot/compressed/Makefile	2019-05-21 02:58:19.920070437 +0200
@@ -24,7 +24,7 @@ OBJECT_FILES_NON_STANDARD	:= y
 KCOV_INSTRUMENT		:= n
 
 targets := vmlinux vmlinux.bin vmlinux.bin.gz vmlinux.bin.bz2 vmlinux.bin.lzma \
-	vmlinux.bin.xz vmlinux.bin.lzo vmlinux.bin.lz4
+	vmlinux.bin.xz vmlinux.bin.lzo vmlinux.bin.lz4 vmlinux.bin.zst
 
 KBUILD_CFLAGS := -m$(BITS) -O2
 KBUILD_CFLAGS += -fno-strict-aliasing $(call cc-option, -fPIE, -fPIC)
@@ -144,6 +144,8 @@ $(obj)/vmlinux.bin.lzo: $(vmlinux.bin.al
 	$(call if_changed,lzo)
 $(obj)/vmlinux.bin.lz4: $(vmlinux.bin.all-y) FORCE
 	$(call if_changed,lz4)
+$(obj)/vmlinux.bin.zst: $(vmlinux.bin.all-y) FORCE
+	$(call if_changed,zstd)
 
 suffix-$(CONFIG_KERNEL_GZIP)	:= gz
 suffix-$(CONFIG_KERNEL_BZIP2)	:= bz2
@@ -151,6 +153,7 @@ suffix-$(CONFIG_KERNEL_LZMA)	:= lzma
 suffix-$(CONFIG_KERNEL_XZ)	:= xz
 suffix-$(CONFIG_KERNEL_LZO) 	:= lzo
 suffix-$(CONFIG_KERNEL_LZ4) 	:= lz4
+suffix-$(CONFIG_KERNEL_ZSTD)	:= zst
 
 quiet_cmd_mkpiggy = MKPIGGY $@
       cmd_mkpiggy = $(obj)/mkpiggy $< > $@
diff -up linux-5.2-rc1/arch/x86/boot/compressed/misc.c.20~ linux-5.2-rc1/arch/x86/boot/compressed/misc.c
--- linux-5.2-rc1/arch/x86/boot/compressed/misc.c.20~	2019-05-20 00:47:09.000000000 +0200
+++ linux-5.2-rc1/arch/x86/boot/compressed/misc.c	2019-05-21 02:58:19.920070437 +0200
@@ -76,6 +76,10 @@ static int lines, cols;
 #ifdef CONFIG_KERNEL_LZ4
 #include "../../../../lib/decompress_unlz4.c"
 #endif
+
+#ifdef CONFIG_KERNEL_ZSTD
+#include "../../../../lib/decompress_unzstd.c"
+#endif
 /*
  * NOTE: When adding a new decompressor, please update the analysis in
  * ../header.S.
diff -up linux-5.2-rc1/arch/x86/boot/header.S.20~ linux-5.2-rc1/arch/x86/boot/header.S
--- linux-5.2-rc1/arch/x86/boot/header.S.20~	2019-05-20 00:47:09.000000000 +0200
+++ linux-5.2-rc1/arch/x86/boot/header.S	2019-05-21 02:58:19.920070437 +0200
@@ -526,8 +526,14 @@ pref_address:		.quad LOAD_PHYSICAL_ADDR
 # the size-dependent part now grows so fast.
 #
 # extra_bytes = (uncompressed_size >> 8) + 65536
+#
+# ZSTD compressed data grows by at most 3 bytes per 128K, and only has a 22
+# byte fixed overhead but has a maximum block size of 128K, so it needs a
+# larger margin.
+#
+# extra_bytes = (uncompressed_size >> 8) + 131072
 
-#define ZO_z_extra_bytes	((ZO_z_output_len >> 8) + 65536)
+#define ZO_z_extra_bytes	((ZO_z_output_len >> 8) + 131072)
 #if ZO_z_output_len > ZO_z_input_len
 # define ZO_z_extract_offset	(ZO_z_output_len + ZO_z_extra_bytes - \
 				 ZO_z_input_len)
diff -up linux-5.2-rc1/arch/x86/include/asm/boot.h.20~ linux-5.2-rc1/arch/x86/include/asm/boot.h
--- linux-5.2-rc1/arch/x86/include/asm/boot.h.20~	2019-05-20 00:47:09.000000000 +0200
+++ linux-5.2-rc1/arch/x86/include/asm/boot.h	2019-05-21 02:58:19.920070437 +0200
@@ -24,9 +24,11 @@
 # error "Invalid value for CONFIG_PHYSICAL_ALIGN"
 #endif
 
-#ifdef CONFIG_KERNEL_BZIP2
+#if defined(CONFIG_KERNEL_BZIP2)
 # define BOOT_HEAP_SIZE		0x400000
-#else /* !CONFIG_KERNEL_BZIP2 */
+#elif defined(CONFIG_KERNEL_ZSTD)
+# define BOOT_HEAP_SIZE		 0x30000
+#else
 # define BOOT_HEAP_SIZE		 0x10000
 #endif
 
diff -up linux-5.2-rc1/arch/x86/Kconfig.20~ linux-5.2-rc1/arch/x86/Kconfig
--- linux-5.2-rc1/arch/x86/Kconfig.20~	2019-05-20 00:47:09.000000000 +0200
+++ linux-5.2-rc1/arch/x86/Kconfig	2019-05-21 02:58:19.921070428 +0200
@@ -172,6 +172,7 @@ config X86
 	select HAVE_KERNEL_LZMA
 	select HAVE_KERNEL_LZO
 	select HAVE_KERNEL_XZ
+	select HAVE_KERNEL_ZSTD
 	select HAVE_KPROBES
 	select HAVE_KPROBES_ON_FTRACE
 	select HAVE_FUNCTION_ERROR_INJECTION
diff -up linux-5.2-rc1/Documentation/x86/boot.rst.20~ linux-5.2-rc1/Documentation/x86/boot.rst
--- linux-5.2-rc1/Documentation/x86/boot.rst.20~	2019-05-21 02:58:19.921070428 +0200
+++ linux-5.2-rc1/Documentation/x86/boot.rst	2019-05-21 02:59:27.597466111 +0200
@@ -768,7 +768,7 @@ Protocol:	2.08+
   numbers.  The currently supported compression formats are gzip
   (magic numbers 1F 8B or 1F 9E), bzip2 (magic number 42 5A), LZMA
   (magic number 5D 00), XZ (magic number FD 37), and LZ4 (magic number
-  02 21).  The uncompressed payload is currently always ELF (magic
+  02 21) and ZSTD (magic number 28 B5).  The uncompressed payload is currently always ELF (magic
   number 7F 45 4C 46).
 
 ============	==============
