--- linux-4.14/drivers/gpu/drm/vboxvideo/vbox_mode.c.omv~	2017-11-21 21:41:24.553971236 +0100
+++ linux-4.14/drivers/gpu/drm/vboxvideo/vbox_mode.c	2017-11-21 21:41:31.119960437 +0100
@@ -402,7 +402,7 @@ static struct drm_encoder *vbox_best_sin
 
 	/* pick the encoder ids */
 	if (enc_id)
-		return drm_encoder_find(connector->dev, enc_id);
+		return drm_encoder_find(connector->dev, NULL, enc_id);
 
 	return NULL;
 }
