diff -up linux-4.17/drivers/pci/vboxpci/linux/VBoxPci-linux.c.omv~ linux-4.17/drivers/pci/vboxpci/linux/VBoxPci-linux.c
--- linux-4.17/drivers/pci/vboxpci/linux/VBoxPci-linux.c.omv~	2018-06-06 21:11:46.934269236 +0200
+++ linux-4.17/drivers/pci/vboxpci/linux/VBoxPci-linux.c	2018-06-06 21:13:34.361275120 +0200
@@ -89,7 +89,11 @@ MODULE_VERSION(VBOX_VERSION_STRING " r"
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 20)
 # define PCI_DEV_GET(v,d,p)            pci_get_device(v,d,p)
 # define PCI_DEV_PUT(x)                pci_dev_put(x)
-# define PCI_DEV_GET_SLOT(bus, devfn)  pci_get_bus_and_slot(bus, devfn)
+# if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 17, 0)
+#  define PCI_DEV_GET_SLOT(bus, devfn)  pci_get_domain_bus_and_slot(0, bus, devfn)
+# else
+#  define PCI_DEV_GET_SLOT(bus, devfn)  pci_get_bus_and_slot(bus, devfn)
+# endif
 #else
 # define PCI_DEV_GET(v,d,p)            pci_find_device(v,d,p)
 # define PCI_DEV_PUT(x)                do { } while (0)
