diff -up linux-5.0-rc2/drivers/virt/vboxdrv/r0drv/linux/memuserkernel-r0drv-linux.c.omv~ linux-5.0-rc2/drivers/virt/vboxdrv/r0drv/linux/memuserkernel-r0drv-linux.c
--- linux-5.0-rc2/drivers/virt/vboxdrv/r0drv/linux/memuserkernel-r0drv-linux.c.omv~	2019-01-14 04:09:04.040288864 +0100
+++ linux-5.0-rc2/drivers/virt/vboxdrv/r0drv/linux/memuserkernel-r0drv-linux.c	2019-01-14 04:11:20.071678928 +0100
@@ -66,7 +66,7 @@ RT_EXPORT_SYMBOL(RTR0MemUserCopyTo);
 RTR0DECL(bool) RTR0MemUserIsValidAddr(RTR3PTR R3Ptr)
 {
     IPRT_LINUX_SAVE_EFL_AC();
-    bool fRc = access_ok(VERIFY_READ, (void *)R3Ptr);
+    bool fRc = access_ok((void *)R3Ptr, sizeof(R3Ptr));
     IPRT_LINUX_RESTORE_EFL_AC();
     return fRc;
 }
diff -up linux-5.0-rc2/fs/vboxsf/super.c.omv~ linux-5.0-rc2/fs/vboxsf/super.c
--- linux-5.0-rc2/fs/vboxsf/super.c.omv~	2019-01-14 04:11:57.486326225 +0100
+++ linux-5.0-rc2/fs/vboxsf/super.c	2019-01-14 04:12:26.646828086 +0100
@@ -12,6 +12,7 @@
 
 #include <linux/magic.h>
 #include <linux/module.h>
+#include <uapi/linux/mount.h>
 #include <linux/nls.h>
 #include <linux/parser.h>
 #include <linux/statfs.h>
