diff -up linux-5.9-rc7/drivers/net/vboxnetadp/r0drv/linux/the-linux-kernel.h.omv~ linux-5.9-rc7/drivers/net/vboxnetadp/r0drv/linux/the-linux-kernel.h
--- linux-5.9-rc7/drivers/net/vboxnetadp/r0drv/linux/the-linux-kernel.h.omv~	2020-09-30 04:13:38.980004985 +0200
+++ linux-5.9-rc7/drivers/net/vboxnetadp/r0drv/linux/the-linux-kernel.h	2020-09-30 04:13:42.703374411 +0200
@@ -137,9 +137,6 @@
 #include <linux/interrupt.h>
 #include <linux/completion.h>
 #include <linux/compiler.h>
-#ifndef HAVE_UNLOCKED_IOCTL /* linux/fs.h defines this */
-# include <linux/smp_lock.h>
-#endif
 /* For the shared folders module */
 #include <linux/vmalloc.h>
 #define wchar_t linux_wchar_t
diff -up linux-5.9-rc7/drivers/net/vboxnetflt/r0drv/linux/the-linux-kernel.h.omv~ linux-5.9-rc7/drivers/net/vboxnetflt/r0drv/linux/the-linux-kernel.h
--- linux-5.9-rc7/drivers/net/vboxnetflt/r0drv/linux/the-linux-kernel.h.omv~	2020-09-30 04:49:22.631654860 +0200
+++ linux-5.9-rc7/drivers/net/vboxnetflt/r0drv/linux/the-linux-kernel.h	2020-09-30 04:49:26.155008745 +0200
@@ -137,9 +137,6 @@
 #include <linux/interrupt.h>
 #include <linux/completion.h>
 #include <linux/compiler.h>
-#ifndef HAVE_UNLOCKED_IOCTL /* linux/fs.h defines this */
-# include <linux/smp_lock.h>
-#endif
 /* For the shared folders module */
 #include <linux/vmalloc.h>
 #define wchar_t linux_wchar_t
diff -up linux-5.9-rc7/drivers/pci/vboxpci/r0drv/linux/the-linux-kernel.h.omv~ linux-5.9-rc7/drivers/pci/vboxpci/r0drv/linux/the-linux-kernel.h
--- linux-5.9-rc7/drivers/pci/vboxpci/r0drv/linux/the-linux-kernel.h.omv~	2020-09-30 03:23:29.781331850 +0200
+++ linux-5.9-rc7/drivers/pci/vboxpci/r0drv/linux/the-linux-kernel.h	2020-09-30 03:23:33.698021525 +0200
@@ -137,9 +137,6 @@
 #include <linux/interrupt.h>
 #include <linux/completion.h>
 #include <linux/compiler.h>
-#ifndef HAVE_UNLOCKED_IOCTL /* linux/fs.h defines this */
-# include <linux/smp_lock.h>
-#endif
 /* For the shared folders module */
 #include <linux/vmalloc.h>
 #define wchar_t linux_wchar_t
diff -up linux-5.9-rc7/drivers/virt/vboxdrv/r0drv/linux/memobj-r0drv-linux.c.omv~ linux-5.9-rc7/drivers/virt/vboxdrv/r0drv/linux/memobj-r0drv-linux.c
--- linux-5.9-rc7/drivers/virt/vboxdrv/r0drv/linux/memobj-r0drv-linux.c.omv~	2020-09-30 13:37:31.971942190 +0200
+++ linux-5.9-rc7/drivers/virt/vboxdrv/r0drv/linux/memobj-r0drv-linux.c	2020-09-30 13:38:07.425359317 +0200
@@ -1146,7 +1146,9 @@ DECLHIDDEN(int) rtR0MemObjNativeLockUser
          */
         else
             rc = get_user_pages_remote(
+# if GET_USER_PAGES_API < KERNEL_VERSION(5, 9, 0)
                                 pTask,                  /* Task for fault accounting. */
+# endif
                                 pTask->mm,              /* Whose pages. */
                                 R3Ptr,                  /* Where from. */
                                 cPages,                 /* How many pages. */
diff -up linux-5.9-rc7/drivers/virt/vboxdrv/r0drv/linux/the-linux-kernel.h.omv~ linux-5.9-rc7/drivers/virt/vboxdrv/r0drv/linux/the-linux-kernel.h
--- linux-5.9-rc7/drivers/virt/vboxdrv/r0drv/linux/the-linux-kernel.h.omv~	2020-09-30 13:28:03.094236249 +0200
+++ linux-5.9-rc7/drivers/virt/vboxdrv/r0drv/linux/the-linux-kernel.h	2020-09-30 13:39:28.022221595 +0200
@@ -137,9 +137,8 @@
 #include <linux/interrupt.h>
 #include <linux/completion.h>
 #include <linux/compiler.h>
-#ifndef HAVE_UNLOCKED_IOCTL /* linux/fs.h defines this */
-# include <linux/smp_lock.h>
-#endif
+/* linux/fs.h defines HAVE_UNLOCKED_IOCTL from 2.6.11 till 5.9, but its meaning remains valid */
+#define HAVE_UNLOCKED_IOCTL 1
 /* For the shared folders module */
 #include <linux/vmalloc.h>
 #define wchar_t linux_wchar_t
diff -up linux-5.9-rc7/drivers/virt/vboxdrv/r0drv/linux/the-linux-kernel.h¨.omv~ linux-5.9-rc7/drivers/virt/vboxdrv/r0drv/linux/the-linux-kernel.h¨
diff -up linux-5.9-rc7/drivers/virt/vboxdrv/r0drv/linux/thread2-r0drv-linux.c.omv~ linux-5.9-rc7/drivers/virt/vboxdrv/r0drv/linux/thread2-r0drv-linux.c
--- linux-5.9-rc7/drivers/virt/vboxdrv/r0drv/linux/thread2-r0drv-linux.c.omv~	2020-10-04 02:28:20.928960825 +0200
+++ linux-5.9-rc7/drivers/virt/vboxdrv/r0drv/linux/thread2-r0drv-linux.c	2020-10-04 02:31:25.949918475 +0200
@@ -78,13 +78,12 @@ DECLHIDDEN(int) rtThreadNativeSetPriorit
             break;
 
         case RTTHREADTYPE_IO:
-            iSchedClass = SCHED_FIFO;
-            Param.sched_priority = MAX_RT_PRIO - 1;
+	    /* Effectively changes prio from 99 to 50 */
+	    sched_set_fifo(current);
             break;
 
         case RTTHREADTYPE_TIMER:
-            iSchedClass = SCHED_FIFO;
-            Param.sched_priority = 1; /* not 0 just in case */
+	    sched_set_fifo_low(current);
             break;
 
         default:
@@ -92,7 +91,9 @@ DECLHIDDEN(int) rtThreadNativeSetPriorit
             return VERR_INVALID_PARAMETER;
     }
 
-    sched_setscheduler(current, iSchedClass, &Param);
+    /* XXX handle SCHED_NORMAL case! */
+    RT_NOREF_PV(iSchedClass);
+    RT_NOREF_PV(Param);
 #else
     RT_NOREF_PV(enmType);
 #endif
