diff -up linux-5.9/drivers/pci/vboxpci/VBoxPciInternal.h.omv~ linux-5.9/drivers/pci/vboxpci/VBoxPciInternal.h
--- linux-5.9/drivers/pci/vboxpci/VBoxPciInternal.h.omv~	2020-10-29 00:08:00.454894538 +0100
+++ linux-5.9/drivers/pci/vboxpci/VBoxPciInternal.h	2020-10-29 00:08:46.181817047 +0100
@@ -34,6 +34,7 @@
 #include <VBox/rawpci.h>
 #include <iprt/semaphore.h>
 #include <iprt/assert.h>
+#include <iprt/linux/version.h>
 
 #ifdef RT_OS_LINUX
 
