--- linux-4.7/drivers/bus/vboxguest/r0drv/linux/timer-r0drv-linux.c.omv~	2016-08-15 14:40:08.673599391 +0200
+++ linux-4.7/drivers/bus/vboxguest/r0drv/linux/timer-r0drv-linux.c	2016-08-15 14:40:32.954446769 +0200
@@ -53,6 +53,8 @@
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 31)
 # define mod_timer_pinned               mod_timer
 # define HRTIMER_MODE_ABS_PINNED        HRTIMER_MODE_ABS
+#elif LINUX_VERSION_CODE >= KERNEL_VERSION(4, 8, 0)
+# define mod_timer_pinned		mod_timer
 #endif
 
 
--- linux-4.7/drivers/gpu/drm/vboxvideo/vbox_ttm.c.omv~	2016-08-15 18:14:24.653098809 +0200
+++ linux-4.7/drivers/gpu/drm/vboxvideo/vbox_ttm.c	2016-08-15 18:16:16.682375627 +0200
@@ -215,6 +215,7 @@ static void vbox_ttm_io_mem_free(struct
 {
 }
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4, 8, 0)
 static int vbox_bo_move(struct ttm_buffer_object *bo,
                bool evict, bool interruptible,
                bool no_wait_gpu,
@@ -224,6 +225,7 @@ static int vbox_bo_move(struct ttm_buffe
     r = ttm_bo_move_memcpy(bo, evict, no_wait_gpu, new_mem);
     return r;
 }
+#endif
 
 
 static void vbox_ttm_backend_destroy(struct ttm_tt *tt)
@@ -270,7 +272,11 @@ struct ttm_bo_driver vbox_bo_driver = {
     .ttm_tt_unpopulate = vbox_ttm_tt_unpopulate,
     .init_mem_type = vbox_bo_init_mem_type,
     .evict_flags = vbox_bo_evict_flags,
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 8, 0)
+    .move = NULL,
+#else
     .move = vbox_bo_move,
+#endif
     .verify_access = vbox_bo_verify_access,
     .io_mem_reserve = &vbox_ttm_io_mem_reserve,
     .io_mem_free = &vbox_ttm_io_mem_free,
--- linux-4.7/drivers/virt/vboxdrv/r0drv/linux/timer-r0drv-linux.c.omv~	2016-08-15 14:29:09.404758055 +0200
+++ linux-4.7/drivers/virt/vboxdrv/r0drv/linux/timer-r0drv-linux.c	2016-08-15 14:39:59.083659678 +0200
@@ -53,6 +53,8 @@
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 31)
 # define mod_timer_pinned               mod_timer
 # define HRTIMER_MODE_ABS_PINNED        HRTIMER_MODE_ABS
+#elif LINUX_VERSION_CODE >= KERNEL_VERSION(4, 8, 0)
+# define mod_timer_pinned		mod_timer
 #endif
 
 
