diff -up linux-5.13/drivers/net/vboxnetadp/r0drv/linux/the-linux-kernel.h.omv~ linux-5.13/drivers/net/vboxnetadp/r0drv/linux/the-linux-kernel.h
--- linux-5.13/drivers/net/vboxnetadp/r0drv/linux/the-linux-kernel.h.omv~	2021-07-09 05:06:23.643468874 +0200
+++ linux-5.13/drivers/net/vboxnetadp/r0drv/linux/the-linux-kernel.h	2021-07-09 05:06:36.147112574 +0200
@@ -42,7 +42,7 @@
 #endif
 #if RT_GNUC_PREREQ(4, 2)
 # pragma GCC diagnostic ignored "-Wunused-parameter"
-# if !defined(__cplusplus) && RT_GNUC_PREREQ(4, 3)
+# if !defined(__cplusplus) && RT_GNUC_PREREQ(4, 3) && !defined(__clang__)
 #  pragma GCC diagnostic ignored "-Wold-style-declaration" /* 2.6.18-411.0.0.0.1.el5/build/include/asm/apic.h:110: warning: 'inline' is not at beginning of declaration [-Wold-style-declaration] */
 # endif
 #endif
diff -up linux-5.13/drivers/virt/vboxdrv/r0drv/linux/the-linux-kernel.h.omv~ linux-5.13/drivers/virt/vboxdrv/r0drv/linux/the-linux-kernel.h
--- linux-5.13/drivers/virt/vboxdrv/r0drv/linux/the-linux-kernel.h.omv~	2021-07-09 16:52:13.756938396 +0200
+++ linux-5.13/drivers/virt/vboxdrv/r0drv/linux/the-linux-kernel.h	2021-07-09 16:52:23.253681655 +0200
@@ -42,7 +42,7 @@
 #endif
 #if RT_GNUC_PREREQ(4, 2)
 # pragma GCC diagnostic ignored "-Wunused-parameter"
-# if !defined(__cplusplus) && RT_GNUC_PREREQ(4, 3)
+# if !defined(__cplusplus) && RT_GNUC_PREREQ(4, 3) && !defined(__clang__)
 #  pragma GCC diagnostic ignored "-Wold-style-declaration" /* 2.6.18-411.0.0.0.1.el5/build/include/asm/apic.h:110: warning: 'inline' is not at beginning of declaration [-Wold-style-declaration] */
 # endif
 #endif
diff -up linux-5.13/drivers/net/vboxnetflt/r0drv/linux/the-linux-kernel.h.omv~ linux-5.13/drivers/net/vboxnetflt/r0drv/linux/the-linux-kernel.h
--- linux-5.13/drivers/net/vboxnetflt/r0drv/linux/the-linux-kernel.h.omv~	2021-07-09 17:22:34.984140145 +0200
+++ linux-5.13/drivers/net/vboxnetflt/r0drv/linux/the-linux-kernel.h	2021-07-09 17:22:42.934198757 +0200
@@ -42,7 +42,7 @@
 #endif
 #if RT_GNUC_PREREQ(4, 2)
 # pragma GCC diagnostic ignored "-Wunused-parameter"
-# if !defined(__cplusplus) && RT_GNUC_PREREQ(4, 3)
+# if !defined(__cplusplus) && RT_GNUC_PREREQ(4, 3) && !defined(__clang__)
 #  pragma GCC diagnostic ignored "-Wold-style-declaration" /* 2.6.18-411.0.0.0.1.el5/build/include/asm/apic.h:110: warning: 'inline' is not at beginning of declaration [-Wold-style-declaration] */
 # endif
 #endif
