diff -up linux-5.10-rc1/kernel/module.c.14~ linux-5.10-rc1/kernel/module.c
--- linux-5.10-rc1/kernel/module.c.14~	2020-10-25 23:14:11.000000000 +0100
+++ linux-5.10-rc1/kernel/module.c	2020-10-26 14:25:23.069595152 +0100
@@ -2094,6 +2094,7 @@ static void module_enable_nx(const struc
 static int module_enforce_rwx_sections(Elf_Ehdr *hdr, Elf_Shdr *sechdrs,
 				       char *secstrings, struct module *mod)
 {
+#if !defined(__aarch64__) // https://lore.kernel.org/lkml/20200403171303.GK20760@hirez.programming.kicks-ass.net/, https://sourceware.org/bugzilla/show_bug.cgi?id=26378
 	const unsigned long shf_wx = SHF_WRITE|SHF_EXECINSTR;
 	int i;
 
@@ -2104,6 +2105,7 @@ static int module_enforce_rwx_sections(E
 			return -ENOEXEC;
 		}
 	}
+#endif
 
 	return 0;
 }
