--- linux-4.14/drivers/staging/rtl8192u/Makefile.0250~	2017-11-12 19:46:13.000000000 +0100
+++ linux-4.14/drivers/staging/rtl8192u/Makefile	2017-12-25 17:45:17.803635299 +0100
@@ -1,7 +1,7 @@
 # SPDX-License-Identifier: GPL-2.0
 NIC_SELECT = RTL8192U
 
-ccflags-y := -std=gnu89
+ccflags-y := -std=gnu11
 ccflags-y += -O2
 
 ccflags-y += -DCONFIG_FORCE_HARD_FLOAT=y
--- linux-4.14/drivers/staging/rtl8723bs/include/ieee80211.h.0250~	2017-11-12 19:46:13.000000000 +0100
+++ linux-4.14/drivers/staging/rtl8723bs/include/ieee80211.h	2017-12-25 17:45:17.803635299 +0100
@@ -1008,18 +1008,18 @@ enum ieee80211_state {
 #define IP_FMT "%pI4"
 #define IP_ARG(x) (x)
 
-extern __inline int is_multicast_mac_addr(const u8 *addr)
+static inline int is_multicast_mac_addr(const u8 *addr)
 {
         return ((addr[0] != 0xff) && (0x01 & addr[0]));
 }
 
-extern __inline int is_broadcast_mac_addr(const u8 *addr)
+static inline int is_broadcast_mac_addr(const u8 *addr)
 {
 	return ((addr[0] == 0xff) && (addr[1] == 0xff) && (addr[2] == 0xff) &&   \
 		(addr[3] == 0xff) && (addr[4] == 0xff) && (addr[5] == 0xff));
 }
 
-extern __inline int is_zero_mac_addr(const u8 *addr)
+static inline int is_zero_mac_addr(const u8 *addr)
 {
 	return ((addr[0] == 0x00) && (addr[1] == 0x00) && (addr[2] == 0x00) &&   \
 		(addr[3] == 0x00) && (addr[4] == 0x00) && (addr[5] == 0x00));
--- linux-4.14/drivers/video/fbdev/i810/i810_dvt.c.0250~	2017-11-12 19:46:13.000000000 +0100
+++ linux-4.14/drivers/video/fbdev/i810/i810_dvt.c	2017-12-25 17:45:17.803635299 +0100
@@ -207,7 +207,7 @@ void round_off_xres(u32 *xres)
 		*xres = 1600;
 }
 
-inline void round_off_yres(u32 *xres, u32 *yres)
+void round_off_yres(u32 *xres, u32 *yres)
 {
 	*yres = (*xres * 3) >> 2;
 }
--- linux-4.14/lib/decompress_unlzo.c.0250~	2017-11-12 19:46:13.000000000 +0100
+++ linux-4.14/lib/decompress_unlzo.c	2017-12-25 17:45:17.803635299 +0100
@@ -52,7 +52,7 @@ static const unsigned char lzop_magic[]
 #define HEADER_SIZE_MIN       (9 + 7     + 4 + 8     + 1       + 4)
 #define HEADER_SIZE_MAX       (9 + 7 + 1 + 8 + 8 + 4 + 1 + 255 + 4)
 
-STATIC inline long INIT parse_header(u8 *input, long *skip, long in_len)
+static inline long INIT parse_header(u8 *input, long *skip, long in_len)
 {
 	int l;
 	u8 *parse = input;
--- linux-4.14/Makefile.0250~	2017-12-25 17:45:17.567633653 +0100
+++ linux-4.14/Makefile	2017-12-25 17:47:27.841791582 +0100
@@ -362,7 +362,7 @@ HOST_LFS_LIBS := $(shell getconf LFS_LIB
 HOSTCC       = gcc
 HOSTCXX      = g++
 HOSTCFLAGS   := -Wall -Wmissing-prototypes -Wstrict-prototypes -O2 \
-		-fomit-frame-pointer -std=gnu89 $(HOST_LFS_CFLAGS)
+		-fomit-frame-pointer -std=gnu11 $(HOST_LFS_CFLAGS)
 HOSTCXXFLAGS := -O2 $(HOST_LFS_CFLAGS)
 HOSTLDFLAGS  := $(HOST_LFS_LDFLAGS)
 HOST_LOADLIBES := $(HOST_LFS_LIBS)
@@ -417,7 +417,7 @@ KBUILD_CFLAGS   := -Wall -Wundef -Wstric
 		   -fno-strict-aliasing -fno-common -fshort-wchar \
 		   -Werror-implicit-function-declaration \
 		   -Wno-format-security \
-		   -std=gnu89
+		   -std=gnu11
 KBUILD_CPPFLAGS := -D__KERNEL__
 KBUILD_AFLAGS_KERNEL :=
 KBUILD_CFLAGS_KERNEL :=
--- linux-4.14/tools/testing/selftests/sync/Makefile.0250~	2017-11-12 19:46:13.000000000 +0100
+++ linux-4.14/tools/testing/selftests/sync/Makefile	2017-12-25 17:45:17.803635299 +0100
@@ -1,5 +1,5 @@
 # SPDX-License-Identifier: GPL-2.0
-CFLAGS += -O2 -g -std=gnu89 -pthread -Wall -Wextra
+CFLAGS += -O2 -g -std=gnu11 -pthread -Wall -Wextra
 CFLAGS += -I../../../../usr/include/
 LDFLAGS += -pthread
 
diff -up linux-4.17/drivers/scsi/lpfc/lpfc_sli4.h.omv~ linux-4.17/drivers/scsi/lpfc/lpfc_sli4.h
--- linux-4.17/drivers/scsi/lpfc/lpfc_sli4.h.omv~	2018-06-06 21:40:30.150363628 +0200
+++ linux-4.17/drivers/scsi/lpfc/lpfc_sli4.h	2018-06-06 21:40:41.153364231 +0200
@@ -866,10 +866,10 @@ void lpfc_sli_remove_dflt_fcf(struct lpf
 int lpfc_sli4_get_els_iocb_cnt(struct lpfc_hba *);
 int lpfc_sli4_get_iocb_cnt(struct lpfc_hba *phba);
 int lpfc_sli4_init_vpi(struct lpfc_vport *);
-inline void lpfc_sli4_eq_clr_intr(struct lpfc_queue *);
+void lpfc_sli4_eq_clr_intr(struct lpfc_queue *);
 uint32_t lpfc_sli4_cq_release(struct lpfc_queue *, bool);
 uint32_t lpfc_sli4_eq_release(struct lpfc_queue *, bool);
-inline void lpfc_sli4_if6_eq_clr_intr(struct lpfc_queue *q);
+void lpfc_sli4_if6_eq_clr_intr(struct lpfc_queue *q);
 uint32_t lpfc_sli4_if6_cq_release(struct lpfc_queue *q, bool arm);
 uint32_t lpfc_sli4_if6_eq_release(struct lpfc_queue *q, bool arm);
 void lpfc_sli4_fcfi_unreg(struct lpfc_hba *, uint16_t);
