diff -up linux-5.0-rc8/drivers/net/ethernet/aquantia/atlantic/hw_atl/hw_atl_b0.c.omv~ linux-5.0-rc8/drivers/net/ethernet/aquantia/atlantic/hw_atl/hw_atl_b0.c
--- linux-5.0-rc8/drivers/net/ethernet/aquantia/atlantic/hw_atl/hw_atl_b0.c.omv~	2019-02-27 00:42:53.540895184 +0100
+++ linux-5.0-rc8/drivers/net/ethernet/aquantia/atlantic/hw_atl/hw_atl_b0.c	2019-02-27 00:43:16.117266512 +0100
@@ -275,6 +275,9 @@ static int hw_atl_b0_hw_offload_set(stru
 
 static int hw_atl_b0_hw_init_tx_path(struct aq_hw_s *self)
 {
+	/* Tx TC/Queue number config */
+	hw_atl_rpb_tps_tx_tc_mode_set(self, 1U);
+
 	hw_atl_thm_lso_tcp_flag_of_first_pkt_set(self, 0x0FF6U);
 	hw_atl_thm_lso_tcp_flag_of_middle_pkt_set(self, 0x0FF6U);
 	hw_atl_thm_lso_tcp_flag_of_last_pkt_set(self, 0x0F7FU);
diff -up linux-5.0-rc8/drivers/net/ethernet/aquantia/atlantic/hw_atl/hw_atl_llh.c.omv~ linux-5.0-rc8/drivers/net/ethernet/aquantia/atlantic/hw_atl/hw_atl_llh.c
--- linux-5.0-rc8/drivers/net/ethernet/aquantia/atlantic/hw_atl/hw_atl_llh.c.omv~	2019-02-27 00:43:25.767425197 +0100
+++ linux-5.0-rc8/drivers/net/ethernet/aquantia/atlantic/hw_atl/hw_atl_llh.c	2019-02-27 00:44:27.310436679 +0100
@@ -1274,6 +1274,15 @@ void hw_atl_tpb_tx_buff_en_set(struct aq
 			    HW_ATL_TPB_TX_BUF_EN_SHIFT, tx_buff_en);
 }
 
+void hw_atl_rpb_tps_tx_tc_mode_set(struct aq_hw_s *aq_hw,
+                                   u32 tx_traf_class_mode)
+{
+	aq_hw_write_reg_bit(aq_hw, HW_ATL_TPB_TX_TC_MODE_ADDR,
+			    HW_ATL_TPB_TX_TC_MODE_MSK,
+			    HW_ATL_TPB_TX_TC_MODE_SHIFT,
+			    tx_traf_class_mode);
+}
+
 void hw_atl_tpb_tx_buff_hi_threshold_per_tc_set(struct aq_hw_s *aq_hw,
 						u32 tx_buff_hi_threshold_per_tc,
 					 u32 buffer)
diff -up linux-5.0-rc8/drivers/net/ethernet/aquantia/atlantic/hw_atl/hw_atl_llh.h.omv~ linux-5.0-rc8/drivers/net/ethernet/aquantia/atlantic/hw_atl/hw_atl_llh.h
--- linux-5.0-rc8/drivers/net/ethernet/aquantia/atlantic/hw_atl/hw_atl_llh.h.omv~	2019-02-27 00:44:40.135647360 +0100
+++ linux-5.0-rc8/drivers/net/ethernet/aquantia/atlantic/hw_atl/hw_atl_llh.h	2019-02-27 00:45:05.930070990 +0100
@@ -605,6 +605,10 @@ void hw_atl_thm_lso_tcp_flag_of_middle_p
 
 /* tpb */
 
+/* set TX Traffic Class Mode */
+void hw_atl_rpb_tps_tx_tc_mode_set(struct aq_hw_s *aq_hw,
+				   u32 tx_traf_class_mode);
+
 /* set tx buffer enable */
 void hw_atl_tpb_tx_buff_en_set(struct aq_hw_s *aq_hw, u32 tx_buff_en);
 
diff -up linux-5.0-rc8/drivers/net/ethernet/aquantia/atlantic/hw_atl/hw_atl_llh_internal.h.omv~ linux-5.0-rc8/drivers/net/ethernet/aquantia/atlantic/hw_atl/hw_atl_llh_internal.h
--- linux-5.0-rc8/drivers/net/ethernet/aquantia/atlantic/hw_atl/hw_atl_llh_internal.h.omv~	2019-02-27 00:45:18.159271784 +0100
+++ linux-5.0-rc8/drivers/net/ethernet/aquantia/atlantic/hw_atl/hw_atl_llh_internal.h	2019-02-27 00:45:37.661591941 +0100
@@ -1948,6 +1948,19 @@
 /* default value of bitfield tx_buf_en */
 #define HW_ATL_TPB_TX_BUF_EN_DEFAULT 0x0
 
+/* register address for bitfield tx_tc_mode */
+#define HW_ATL_TPB_TX_TC_MODE_ADDR 0x00007900
+/* bitmask for bitfield tx_tc_mode */
+#define HW_ATL_TPB_TX_TC_MODE_MSK 0x00000100
+/* inverted bitmask for bitfield tx_tc_mode */
+#define HW_ATL_TPB_TX_TC_MODE_MSKN 0xFFFFFEFF
+/* lower bit position of bitfield tx_tc_mode */
+#define HW_ATL_TPB_TX_TC_MODE_SHIFT 8
+/* width of bitfield tx_tc_mode */
+#define HW_ATL_TPB_TX_TC_MODE_WIDTH 1
+/* default value of bitfield tx_tc_mode */
+#define HW_ATL_TPB_TX_TC_MODE_DEFAULT 0x0
+
 /* tx tx{b}_hi_thresh[c:0] bitfield definitions
  * preprocessor definitions for the bitfield "tx{b}_hi_thresh[c:0]".
  * parameter: buffer {b} | stride size 0x10 | range [0, 7]
